<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Kaixuan Luo" />


<title>Tutorial for making track plots to visualize finemapping results for genomic regions</title>

<script src="site_libs/header-attrs-2.20/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Analysis pipelines</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/kevinlkx/analysis_pipelines">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Tutorial for making track plots to
visualize finemapping results for genomic regions</h1>
<h4 class="author">Kaixuan Luo</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-10-20
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>analysis_pipelines/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200524code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20200524)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200524code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200524)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomkevinlkxanalysispipelinestree345fe1926ebd15257259236a5ce8dbad725f124atargetblank345fe19a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/kevinlkx/analysis_pipelines/tree/345fe1926ebd15257259236a5ce8dbad725f124a" target="_blank">345fe19</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomkevinlkxanalysispipelinestree345fe1926ebd15257259236a5ce8dbad725f124atargetblank345fe19a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/kevinlkx/analysis_pipelines/tree/345fe1926ebd15257259236a5ce8dbad725f124a" target="_blank">345fe19</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  analysis/test_sldsc_splicingAnnot.Rmd
    Untracked:  code/compute_ldscore_generic_annot.sbatch
    Untracked:  code/extract_baselineLD_generic_annot.R
    Untracked:  code/ldsc_make_binary_annot_compute_ldscores_bedfiles.sbatch
    Untracked:  code/make_ldsc_binary_annots_from_bedfiles.R
    Untracked:  code/sldsc_annot_generic_baselineLD_separate.sbatch
    Untracked:  scripts/tmp.R

Unstaged changes:
    Modified:   analysis/index.Rmd
    Modified:   analysis/mapgen_torus_enrichment_heart_atlas.Rmd
    Modified:   analysis/mapgen_torus_susie_AF.Rmd
    Modified:   analysis/sldsc_example_GTEx_QTLs.Rmd
    Modified:   analysis/sldsc_pipeline.Rmd
    Modified:   code/extract_baselineLDv2.2_generic_annot.R
    Modified:   code/mapgen_trackplots.R
    Modified:   scripts/run_finemapping.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/make_mapgen_trackplots.Rmd</code>) and HTML
(<code>docs/make_mapgen_trackplots.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/kevinlkx/analysis_pipelines/blob/345fe1926ebd15257259236a5ce8dbad725f124a/analysis/make_mapgen_trackplots.Rmd" target="_blank">345fe19</a>
</td>
<td>
kevinlkx
</td>
<td>
2023-10-20
</td>
<td>
wflow_publish("analysis/make_mapgen_trackplots.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/kevinlkx/analysis_pipelines/f780ba6941a99169563b7312e179e986e6035986/docs/make_mapgen_trackplots.html" target="_blank">f780ba6</a>
</td>
<td>
kevinlkx
</td>
<td>
2022-09-12
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/kevinlkx/analysis_pipelines/blob/a014500a8df84c62b548518ccba8abef8aa9a870/analysis/make_mapgen_trackplots.Rmd" target="_blank">a014500</a>
</td>
<td>
kevinlkx
</td>
<td>
2022-09-12
</td>
<td>
added ABC data
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/kevinlkx/analysis_pipelines/da4ec13806863fd03b668f51aeefa6790355bf22/docs/make_mapgen_trackplots.html" target="_blank">da4ec13</a>
</td>
<td>
kevinlkx
</td>
<td>
2022-09-12
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/kevinlkx/analysis_pipelines/blob/adbc5ca0e412b3b281ca1e4d1467959c23294899/analysis/make_mapgen_trackplots.Rmd" target="_blank">adbc5ca</a>
</td>
<td>
kevinlkx
</td>
<td>
2022-09-12
</td>
<td>
added ABC data
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/kevinlkx/analysis_pipelines/ff411beb527b6cb770e90dd71d472b35084d8b2c/docs/make_mapgen_trackplots.html" target="_blank">ff411be</a>
</td>
<td>
kevinlkx
</td>
<td>
2022-09-12
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/kevinlkx/analysis_pipelines/blob/2b0f109d0555c92eb8adb643ebf5d38ed99f5d88/analysis/make_mapgen_trackplots.Rmd" target="_blank">2b0f109</a>
</td>
<td>
kevinlkx
</td>
<td>
2022-09-12
</td>
<td>
updated trackplot with gtf based txdb
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/kevinlkx/analysis_pipelines/76c15461ee95c6ec35ecaefc1dadf5e86d609f30/docs/make_mapgen_trackplots.html" target="_blank">76c1546</a>
</td>
<td>
kevinlkx
</td>
<td>
2022-09-12
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/kevinlkx/analysis_pipelines/blob/cdf3452d422854214cafe52e81504e73821adb30/analysis/make_mapgen_trackplots.Rmd" target="_blank">cdf3452</a>
</td>
<td>
kevinlkx
</td>
<td>
2022-09-12
</td>
<td>
added txdb built from gtf
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/kevinlkx/analysis_pipelines/45bb1945a94a9eb3eb00e0b0ec4cec6694834c3a/docs/make_mapgen_trackplots.html" target="_blank">45bb194</a>
</td>
<td>
kevinlkx
</td>
<td>
2022-08-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/kevinlkx/analysis_pipelines/blob/6e9c89d42def2c21555975876f81a0f4c2c86deb/analysis/make_mapgen_trackplots.Rmd" target="_blank">6e9c89d</a>
</td>
<td>
kevinlkx
</td>
<td>
2022-08-19
</td>
<td>
added a note about counts data format
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/kevinlkx/analysis_pipelines/ca16d13ab9a8d0d8cacb2e2f0f207664323f010c/docs/make_mapgen_trackplots.html" target="_blank">ca16d13</a>
</td>
<td>
kevinlkx
</td>
<td>
2022-08-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/kevinlkx/analysis_pipelines/blob/f05e951b488009fae10ab7f6d2ef38879c69305b/analysis/make_mapgen_trackplots.Rmd" target="_blank">f05e951</a>
</td>
<td>
kevinlkx
</td>
<td>
2022-08-18
</td>
<td>
added a tutorial for making track plots
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/kevinlkx/analysis_pipelines/blob/5295f532194a434bc0338f619e1ec9a4d5215956/analysis/make_mapgen_trackplots.Rmd" target="_blank">5295f53</a>
</td>
<td>
kevinlkx
</td>
<td>
2022-08-18
</td>
<td>
added trackplot tutorials
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="track-plot-tutorial" class="section level1">
<h1>Track plot tutorial</h1>
<p><strong>Required input data</strong>:</p>
<ul>
<li>Genetic fine-mapping summary statistics.</li>
<li>Gene annotations (exons, introns, UTRs, etc.).</li>
<li>Functional annotation data, e.g.: ATAC-seq data, histone ChIP-seq
peaks, PC-HiC loops, etc.</li>
</ul>
<p>To make the trackplots, you will need to have the following packages
installed: <code>GenomicFeatures</code>, <code>AnnotationDbi</code>,
<code>org.Hs.eg.db</code>, <code>GenomicInteractions</code>,
<code>Gviz</code>, <code>rtracklayer</code> from Bioconductor.</p>
<p>Load R packages</p>
<pre class="r"><code>library(GenomicFeatures) #  Making and manipulating annotations
library(rtracklayer) # Import annotation data
library(Gviz) # R package used to visualize track plots
library(GenomicInteractions) # visualize HiC loops
library(AnnotationDbi) # match gene ID to gene symbol
library(org.Hs.eg.db) # match gene ID to gene symbol
library(mapgen)</code></pre>
<pre class="r"><code>trackdata.dir &lt;- &quot;/project2/xinhe/shared_data/mapgen/example_data/trackplot&quot;</code></pre>
<p>Load fine-mapping results.</p>
<pre class="r"><code>finemapstats &lt;- readRDS(system.file(&quot;extdata&quot;, &quot;AF.finemapping.sumstats.rds&quot;, package = &quot;mapgen&quot;))
finemapstats &lt;- process_finemapping_sumstats(finemapstats, 
                                             snp = &#39;snp&#39;, 
                                             chr = &#39;chr&#39;, 
                                             pos = &#39;pos&#39;, 
                                             pip = &#39;susie_pip&#39;, 
                                             pval = &#39;pval&#39;, 
                                             zscore = &#39;zscore&#39;, 
                                             cs = &#39;cs&#39;, 
                                             locus = &#39;locus&#39;,  
                                             pip.thresh = 0)</code></pre>
<pre><code>Processing fine-mapping summary statistics ...</code></pre>
<p>Load genomic annotations and gene information</p>
<p>We included gene annotations (hg19) in the package, downloaded from
<a
href="https://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_19/gencode.v19.annotation.gtf.gz">GENCODE
release 19</a>.</p>
<pre class="r"><code>genomic.annots &lt;- readRDS(system.file(&quot;extdata&quot;, &quot;genomic.annots.hg19.rds&quot;, package = &quot;mapgen&quot;))
gene.annots &lt;- genomic.annots$genes</code></pre>
<p>Load Promoter-capture HiC (PCHi-C) data from iPSC derived
cardiomyocytes (CMs).</p>
<pre class="r"><code>pcHiC &lt;- readRDS(system.file(&quot;extdata&quot;, &quot;pcHiC.CM.gr.rds&quot;, package = &quot;mapgen&quot;))
pcHiC &lt;- pcHiC[pcHiC$gene_name %in% gene.annots$gene_name, ] # restrict to protein coding genes</code></pre>
<p>Load ABC data</p>
<pre class="r"><code>ABC &lt;- data.table::fread(system.file(&quot;extdata&quot;, &quot;heart_ventricle-ENCODE_ABC.tsv.gz&quot;, package = &quot;mapgen&quot;))
ABC &lt;- process_ABC(ABC, full.element = TRUE)
ABC &lt;- ABC[ABC$gene_name %in% gene.annots$gene_name, ] # restrict to protein coding genes
ABC$score &lt;- ABC$score * 100 # scale to visualize the ABC scores
head(ABC, 3)</code></pre>
<pre><code>GRanges object with 3 ranges and 4 metadata columns:
      seqnames        ranges strand | promoter_start promoter_end   gene_name
         &lt;Rle&gt;     &lt;IRanges&gt;  &lt;Rle&gt; |      &lt;integer&gt;    &lt;integer&gt; &lt;character&gt;
  [1]     chr1 888243-888743      * |         894679       894679       NOC2L
  [2]     chr1 908361-908861      * |         895966       895966      KLHL17
  [3]     chr1 908361-908861      * |         901876       901876     PLEKHN1
          score
      &lt;numeric&gt;
  [1]    1.5224
  [2]    1.7673
  [3]    4.1100
  -------
  seqinfo: 23 sequences from an unspecified genome; no seqlengths</code></pre>
<p>Load H3K27ac and DHS bed files</p>
<pre class="r"><code>H3K27ac_peaks &lt;- rtracklayer::import(file.path(trackdata.dir, &quot;H3K27ac.heart.concat.hg19.bed.gz&quot;))
DHS_peaks &lt;- rtracklayer::import(file.path(trackdata.dir, &quot;FetalHeart_E083.DNase.hg19.narrowPeak.bed.gz&quot;))</code></pre>
<p>Load ATAC data files. These data need to be in wig, bigWig/bw,
bedGraph, or bam format.</p>
<pre class="r"><code>CM_counts &lt;- rtracklayer::import(file.path(trackdata.dir, &quot;Cardiomyocyte.atac.hg19.bedGraph.gz&quot;))
Endo_counts &lt;- rtracklayer::import(file.path(trackdata.dir, &quot;Endothelial.atac.hg19.bedGraph.gz&quot;)) 
Fibro_counts &lt;- rtracklayer::import(file.path(trackdata.dir, &quot;Fibroblast.atac.hg19.bedGraph.gz&quot;))</code></pre>
<p>You can build a <code>TxDb</code> database (“.sqlite”) using gene
annotations (GTF format) from <a
href="https://www.gencodegenes.org">GENCODE</a>, and use to use the
<code>TxDb</code> database.</p>
<pre class="r"><code>txdb &lt;- makeTxDbFromGFF(file.path(trackdata.dir, &#39;gencode.v19.annotation.gtf.gz&#39;), format = &quot;gtf&quot;)
saveDb(txdb, file.path(trackdata.dir, &quot;gencode.v19.annotation.gtf.sqlite&quot;))</code></pre>
<p>If you are in Xin He lab at UChicago, you can access the gene
annotations and <code>TxDb</code> database in the directory:
<code>/project2/xinhe/shared_data/gencode/</code> from RCC.</p>
<pre class="r"><code>txdb &lt;- loadDb(&quot;/project2/xinhe/shared_data/gencode/gencode.v19.annotation.gtf.sqlite&quot;)</code></pre>
<p>You will need the <code>bigsnpr</code> package if you want to
visualize R^2 between SNPs using a reference panel in
<code>bigSNP</code> object.</p>
<p>We provided a <code>bigSNP</code> object of the reference genotype
panel from the 1000 Genomes (1KG) European population.</p>
<p>If you are in the He lab at UChicago, you can load the
<code>bigSNP</code> object from RCC as below.</p>
<p>We use a reference genotype panel from European population (1KG).</p>
<pre class="r"><code>library(bigsnpr) # loading reference genotype for LD calculation</code></pre>
<pre><code>Loading required package: bigstatsr</code></pre>
<pre class="r"><code>bigSNP &lt;- bigsnpr::snp_attach(rdsfile = &#39;/project2/xinhe/1kg/bigsnpr/EUR_variable_1kg.rds&#39;)</code></pre>
<pre><code>FBM from an old version? Reconstructing..</code></pre>
<pre><code>You should use `snp_save()`.</code></pre>
<div id="make-track-plots" class="section level2">
<h2>Make track plots</h2>
<p>Plot HCN4 locus in the genomic region “chr15:73610000-73700000”</p>
<p>Highlight top SNP (“rs7172038”)</p>
<pre class="r"><code>counts &lt;- list(&quot;CM&quot; = CM_counts, &quot;Endo&quot; = Endo_counts, &quot;Fibro&quot; = Fibro_counts)
peaks &lt;- list(&quot;H3K27ac&quot; = H3K27ac_peaks, &quot;DHS&quot; = DHS_peaks)
loops &lt;- list(&quot;ABC&quot; = ABC)

track_plot(finemapstats,
           region = &quot;chr15:73610000-73700000&quot;,
           gene.annots,
           bigSNP = bigSNP,
           txdb = txdb,
           counts = counts,
           peaks = peaks,
           loops = loops,
           genome = &quot;hg19&quot;,
           filter_loop_genes = &quot;HCN4&quot;,
           highlight_snps = &quot;topSNP&quot;, 
           counts.color = c(&quot;red&quot;, &quot;green&quot;, &quot;purple&quot;),
           peaks.color = c(&quot;navy&quot;, &quot;blue&quot;),
           loops.color = &quot;gray&quot;, 
           genelabel.side = &quot;above&quot;,
           verbose = TRUE)</code></pre>
<pre><code>463 snps included.
Color SNPs in PIP track by loci. 
Adding CM track...
Adding Endo track...
Adding Fibro track...
Adding H3K27ac track...
Adding DHS track...
Adding ABC track...
Only show ABC loops linked to gene: HCN4 
Making gene track object using txdb database ...
Highlight SNP: rs7172038 
Making track plot ...</code></pre>
<p><img src="figure/make_mapgen_trackplots.Rmd/make-HCN4-trackplot-1.png" width="1152" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
 [1] LC_CTYPE=en_US.UTF-8 LC_NUMERIC=C         LC_TIME=C           
 [4] LC_COLLATE=C         LC_MONETARY=C        LC_MESSAGES=C       
 [7] LC_PAPER=C           LC_NAME=C            LC_ADDRESS=C        
[10] LC_TELEPHONE=C       LC_MEASUREMENT=C     LC_IDENTIFICATION=C 

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] bigsnpr_1.11.6              bigstatsr_1.5.12           
 [3] mapgen_0.5.6                org.Hs.eg.db_3.15.0        
 [5] GenomicInteractions_1.32.0  InteractionSet_1.26.1      
 [7] SummarizedExperiment_1.28.0 MatrixGenerics_1.10.0      
 [9] matrixStats_0.63.0          Gviz_1.42.0                
[11] rtracklayer_1.58.0          GenomicFeatures_1.50.4     
[13] AnnotationDbi_1.60.0        Biobase_2.58.0             
[15] GenomicRanges_1.48.0        GenomeInfoDb_1.34.9        
[17] IRanges_2.32.0              S4Vectors_0.36.1           
[19] BiocGenerics_0.44.0         workflowr_1.7.0            

loaded via a namespace (and not attached):
  [1] utf8_1.2.3               R.utils_2.12.2           tidyselect_1.2.0        
  [4] RSQLite_2.2.20           htmlwidgets_1.6.1        BiocParallel_1.32.5     
  [7] munsell_0.5.0            codetools_0.2-18         interp_1.1-3            
 [10] withr_2.5.0              colorspace_2.1-0         filelock_1.0.2          
 [13] highr_0.10               knitr_1.42               rstudioapi_0.14         
 [16] git2r_0.30.1             GenomeInfoDbData_1.2.9   bit64_4.0.5             
 [19] rprojroot_2.0.3          vctrs_0.5.2              generics_0.1.3          
 [22] xfun_0.37                biovizBase_1.46.0        timechange_0.2.0        
 [25] BiocFileCache_2.6.0      R6_2.5.1                 bigassertr_0.1.6        
 [28] doParallel_1.0.17        bigsparser_0.6.1         AnnotationFilter_1.22.0 
 [31] bitops_1.0-7             cachem_1.0.6             DelayedArray_0.24.0     
 [34] assertthat_0.2.1         promises_1.2.0.1         BiocIO_1.8.0            
 [37] scales_1.2.1             nnet_7.3-17              googlesheets4_1.0.1     
 [40] gtable_0.3.1             processx_3.8.0           ensembldb_2.22.0        
 [43] rlang_1.0.6              flock_0.7                splines_4.2.0           
 [46] lazyeval_0.2.2           gargle_1.3.0             dichromat_2.0-0.1       
 [49] broom_1.0.3              plyranges_1.18.0         checkmate_2.1.0         
 [52] yaml_2.3.7               bigparallelr_0.3.2       modelr_0.1.10           
 [55] backports_1.4.1          httpuv_1.6.5             Hmisc_4.8-0             
 [58] tools_4.2.0              ggplot2_3.4.1            ellipsis_0.3.2          
 [61] jquerylib_0.1.4          RColorBrewer_1.1-3       Rcpp_1.0.10             
 [64] base64enc_0.1-3          progress_1.2.2           zlibbioc_1.44.0         
 [67] purrr_1.0.1              RCurl_1.98-1.10          ps_1.7.2                
 [70] prettyunits_1.1.1        rpart_4.1.16             deldir_1.0-6            
 [73] cowplot_1.1.1            haven_2.5.1              cluster_2.1.3           
 [76] fs_1.6.1                 magrittr_2.0.3           data.table_1.14.6       
 [79] reprex_2.0.2             googledrive_2.0.0        whisker_0.4             
 [82] ProtGenerics_1.30.0      hms_1.1.2                evaluate_0.20           
 [85] XML_3.99-0.13            jpeg_0.1-10              readxl_1.4.2            
 [88] gridExtra_2.3            compiler_4.2.0           biomaRt_2.54.0          
 [91] tibble_3.1.8             crayon_1.5.2             R.oo_1.25.0             
 [94] htmltools_0.5.4          later_1.3.0              tzdb_0.3.0              
 [97] Formula_1.2-4            tidyr_1.3.0              lubridate_1.9.2         
[100] DBI_1.1.3                dbplyr_2.3.0             rappdirs_0.3.3          
[103] Matrix_1.5-3             readr_2.1.4              cli_3.6.0               
[106] R.methodsS3_1.8.2        parallel_4.2.0           igraph_1.4.0            
[109] forcats_1.0.0            pkgconfig_2.0.3          getPass_0.2-2           
[112] GenomicAlignments_1.34.0 foreign_0.8-82           xml2_1.3.3              
[115] foreach_1.5.2            bslib_0.4.2              rngtools_1.5.2          
[118] XVector_0.38.0           rvest_1.0.3              doRNG_1.8.6             
[121] stringr_1.5.0            VariantAnnotation_1.44.1 callr_3.7.3             
[124] digest_0.6.31            Biostrings_2.66.0        rmarkdown_2.20          
[127] cellranger_1.1.0         htmlTable_2.4.1          restfulr_0.0.15         
[130] curl_5.0.0               Rsamtools_2.12.0         rjson_0.2.21            
[133] lifecycle_1.0.3          jsonlite_1.8.4           BSgenome_1.66.2         
[136] fansi_1.0.4              pillar_1.8.1             lattice_0.20-45         
[139] KEGGREST_1.38.0          fastmap_1.1.0            httr_1.4.4              
[142] survival_3.3-1           glue_1.6.2               png_0.1-8               
[145] iterators_1.0.14         bit_4.0.5                stringi_1.7.12          
[148] sass_0.4.5               blob_1.2.3               latticeExtra_0.6-30     
[151] memoise_2.0.1            dplyr_1.1.0              tidyverse_1.3.2         </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
