<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Kaixuan Luo" />

<meta name="date" content="2020-07-08" />

<title>S-LDSC example analysis to estimate the enrichment of GWAS variants in neuron ATAC-seq OCR and ASoC variants</title>

<script src="site_libs/jquery-1.12.4/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.5.1/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.13/datatables.js"></script>
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.0.0/css/crosstalk.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.0.0/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Analysis pipelines</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/kevinlkx/analysis_pipelines">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">S-LDSC example analysis to estimate the enrichment of GWAS variants in neuron ATAC-seq OCR and ASoC variants</h1>
<h4 class="author">Kaixuan Luo</h4>
<h4 class="date">07/08/2020</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-07-08
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>analysis_pipelines/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200524code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200524)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200524code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200524)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomkevinlkxanalysispipelinestreeaa4aa09cc4ee1bd669b95e24deb466f8bb5edbd8targetblankaa4aa09a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/kevinlkx/analysis_pipelines/tree/aa4aa09cc4ee1bd669b95e24deb466f8bb5edbd8" target="_blank">aa4aa09</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomkevinlkxanalysispipelinestreeaa4aa09cc4ee1bd669b95e24deb466f8bb5edbd8targetblankaa4aa09a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/kevinlkx/analysis_pipelines/tree/aa4aa09cc4ee1bd669b95e24deb466f8bb5edbd8" target="_blank">aa4aa09</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Unstaged changes:
    Modified:   analysis/sldsc_example_GTEx_QTLs.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/sldsc_example_neuron_ATACseq.Rmd</code>) and HTML (<code>docs/sldsc_example_neuron_ATACseq.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/kevinlkx/analysis_pipelines/blob/aa4aa09cc4ee1bd669b95e24deb466f8bb5edbd8/analysis/sldsc_example_neuron_ATACseq.Rmd" target="_blank">aa4aa09</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-07-08
</td>
<td>
wflow_publish(“analysis/sldsc_example_neuron_ATACseq.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/kevinlkx/analysis_pipelines/5c9cebf62c508f99d97b117e98d51b2ac755c4ce/docs/sldsc_example_neuron_ATACseq.html" target="_blank">5c9cebf</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-07-08
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/kevinlkx/analysis_pipelines/4d8d289ae9d4b854f3b7e905d030b80f4a8a2bed/docs/sldsc_example_neuron_ATACseq.html" target="_blank">4d8d289</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-07-08
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/kevinlkx/analysis_pipelines/blob/58daf6fb19a9ed2d2f8517bb8a8def141cf4ae57/analysis/sldsc_example_neuron_ATACseq.Rmd" target="_blank">58daf6f</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-07-08
</td>
<td>
wflow_publish(“analysis/sldsc_example_neuron_ATACseq.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/kevinlkx/analysis_pipelines/03346651e2081f52eb9fe4bdf16d120d3233180b/docs/sldsc_example_neuron_ATACseq.html" target="_blank">0334665</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-07-08
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/kevinlkx/analysis_pipelines/blob/e2d790426655bd655623b2ae253f98cfcab816c0/analysis/sldsc_example_neuron_ATACseq.Rmd" target="_blank">e2d7904</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-07-08
</td>
<td>
wflow_publish(“analysis/sldsc_example_neuron_ATACseq.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/kevinlkx/analysis_pipelines/271e3417479a28aee62163ee706ef224c93adad4/docs/sldsc_example_neuron_ATACseq.html" target="_blank">271e341</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-07-08
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/kevinlkx/analysis_pipelines/blob/54180eeb81250bd77554f568baa1fd8f676f3c5a/analysis/sldsc_example_neuron_ATACseq.Rmd" target="_blank">54180ee</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-07-08
</td>
<td>
wflow_publish(“analysis/sldsc_example_neuron_ATACseq.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/kevinlkx/analysis_pipelines/688b484bf30f6f079083248d8001a570efae3a68/docs/sldsc_example_neuron_ATACseq.html" target="_blank">688b484</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-07-08
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/kevinlkx/analysis_pipelines/blob/43a09532ca80425aad80927be9c7e0988e6b9a71/analysis/sldsc_example_neuron_ATACseq.Rmd" target="_blank">43a0953</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-07-08
</td>
<td>
wflow_publish(“analysis/sldsc_example_neuron_ATACseq.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/kevinlkx/analysis_pipelines/d04827965bcedf5362d084bce7f6139a0c6c4dbf/docs/sldsc_example_neuron_ATACseq.html" target="_blank">d048279</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-07-08
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/kevinlkx/analysis_pipelines/blob/e83e68263d8136af4935fd3e90b553ef88875489/analysis/sldsc_example_neuron_ATACseq.Rmd" target="_blank">e83e682</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-07-08
</td>
<td>
wflow_publish(“analysis/sldsc_example_neuron_ATACseq.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/kevinlkx/analysis_pipelines/588df2f4638516c0cf1e465620990a2705c9625b/docs/sldsc_example_neuron_ATACseq.html" target="_blank">588df2f</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-07-08
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/kevinlkx/analysis_pipelines/blob/2588e9d1342d3f998ab894e1f26ba10206f174fe/analysis/sldsc_example_neuron_ATACseq.Rmd" target="_blank">2588e9d</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-07-08
</td>
<td>
wflow_publish(“analysis/sldsc_example_neuron_ATACseq.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/kevinlkx/analysis_pipelines/1ba32ea4f1cb2ef888e47b5306420fa29d7dc555/docs/sldsc_example_neuron_ATACseq.html" target="_blank">1ba32ea</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-07-08
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/kevinlkx/analysis_pipelines/blob/97bc801da127cf88d3a40d2e5dc2f5ad5c43bb5b/analysis/sldsc_example_neuron_ATACseq.Rmd" target="_blank">97bc801</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-07-08
</td>
<td>
wflow_publish(“analysis/sldsc_example_neuron_ATACseq.Rmd”)
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>This tutorial shows an example analysis using S-LDSC analysis to estimate the enrichment of GWAS variants in neuron ATAC-seq annotations: open chromatin regions (OCR) and allele-specific open chromatin (ASoC) variants. The analysis uses data from our paper: Zhang et al. Allele-specific open chromatin in human iPSC neurons elucidates functional noncoding disease variants. Science 2020.</p>
<div id="prepare-annotation-files-and-compute-ld-scores-for-annotations" class="section level2">
<h2>Prepare annotation files and compute LD scores for annotations</h2>
<div id="prepare-annotations-in-bed-format-for-asoc-binary-annotations" class="section level3">
<h3>Prepare annotations in BED format for ASoC binary annotations</h3>
<ul>
<li>Convert ASoC binary annotations to BED format. The annotations can be found in <code>/project2/xinhe/kevinluo/ldsc/annot/annot_bed/</code></li>
<li>Annotations for ATAC-seq peaks in BED format can also be found in <code>/project2/xinhe/kevinluo/ldsc/annot/annot_bed/</code>.</li>
</ul>
<pre class="r"><code>## Prepare ASoC binary annotations in BED format for LDSC analysis
dir_annot_bed &lt;- &quot;/project2/xinhe/kevinluo/ldsc/annot/annot_bed/&quot;

### ASoC_glut_anno_hg19
annot_name &lt;- &quot;ASoC_glut_anno_hg19&quot;
annot_filename &lt;- &quot;ASoC_glut_anno_hg19.txt&quot;
ASoC_annot &lt;- read.table(paste0(dir_annot_bed, &quot;/&quot;, annot_filename), header = F, stringsAsFactors = F)
colnames(ASoC_annot) &lt;- c(&quot;chr&quot;, &quot;SNP_POS&quot;, &quot;annot&quot;)

ASoC_sig &lt;- ASoC_annot[ASoC_annot$annot == 1, ]
ASoC_sig.bed &lt;- data.frame(chr = ASoC_sig$chr, start = ASoC_sig$SNP_POS - 1, end = ASoC_sig$SNP_POS)
ASoC_sig.bed$chr &lt;- factor(ASoC_sig.bed$chr, levels = paste0(&quot;chr&quot;, 1:22))
ASoC_sig.bed &lt;- ASoC_sig.bed[order(ASoC_sig.bed$chr, ASoC_sig.bed$start), ]
ASoC_sig.bed &lt;- unique(ASoC_sig.bed)
cat(nrow(ASoC_sig.bed), &quot;SNPs with annotation:&quot;, annot_name, &quot;\n&quot;)
write.table(ASoC_sig.bed, paste0(dir_annot_bed, &quot;/&quot;, annot_name, &quot;.bed&quot;), sep = &quot;\t&quot;, col.names = F, row.names = F, quote = F)

### ASoC_npc_anno_hg19
annot_name &lt;- &quot;ASoC_npc_anno_hg19&quot;
annot_filename &lt;- &quot;ASoC_npc_anno_hg19.txt&quot;
ASoC_annot &lt;- read.table(paste0(dir_annot_bed, &quot;/&quot;, annot_filename), header = F, stringsAsFactors = F)
colnames(ASoC_annot) &lt;- c(&quot;chr&quot;, &quot;SNP_POS&quot;, &quot;annot&quot;)

ASoC_sig &lt;- ASoC_annot[ASoC_annot$annot == 1, ]
ASoC_sig.bed &lt;- data.frame(chr = ASoC_sig$chr, start = ASoC_sig$SNP_POS - 1, end = ASoC_sig$SNP_POS)
ASoC_sig.bed$chr &lt;- factor(ASoC_sig.bed$chr, levels = paste0(&quot;chr&quot;, 1:22))
ASoC_sig.bed &lt;- ASoC_sig.bed[order(ASoC_sig.bed$chr, ASoC_sig.bed$start), ]
ASoC_sig.bed &lt;- unique(ASoC_sig.bed)
cat(nrow(ASoC_sig.bed), &quot;SNPs with annotation:&quot;, annot_name, &quot;\n&quot;)
write.table(ASoC_sig.bed, paste0(dir_annot_bed, &quot;/&quot;, annot_name, &quot;.bed&quot;), sep = &quot;\t&quot;, col.names = F, row.names = F, quote = F)</code></pre>
</div>
<div id="compute-ld-scores-for-atac-seq-peaks-and-asoc-annotations" class="section level3">
<h3>Compute LD scores for ATAC-seq peaks and ASoC annotations</h3>
<p>The following <a href="https://github.com/kevinlkx/analysis_pipelines/tree/master/code/ldsc_binary_annot_QTL.sbatch">code</a> generates ldsc-friendly annotation files (annot.gz) from the annotation BED files, then computes LD scores with the annot file (annot.gz).</p>
<pre class="bash"><code>## Compute LD scores for ATAC-seq peak annotations
dir_code=~/projects/analysis_pipelines/code/

sbatch ${dir_code}/ldsc_binary_annot_QTL.sbatch CN_all_peaks.narrowPeak.cleaned.hg19.merged

sbatch ${dir_code}/ldsc_binary_annot_QTL.sbatch DN_all_peaks.narrowPeak.cleaned.hg19.merged

sbatch ${dir_code}/ldsc_binary_annot_QTL.sbatch GA_all_peaks.narrowPeak.cleaned.hg19.merged

sbatch ${dir_code}/ldsc_binary_annot_QTL.sbatch ips_all_peaks.narrowPeak.cleaned.hg19.merged

sbatch ${dir_code}/ldsc_binary_annot_QTL.sbatch NSC_all_peaks.narrowPeak.cleaned.hg19.merged

sbatch ${dir_code}/ldsc_binary_annot_QTL.sbatch NSC_all_peaks.narrowPeak.cleaned.hg19.merged

## Compute LD scores for ASoC annotations
sbatch ${dir_code}/ldsc_binary_annot_QTL.sbatch ASoC_glut_anno_hg19

sbatch ${dir_code}/ldsc_binary_annot_QTL.sbatch ASoC_npc_anno_hg19
</code></pre>
<p>Computed LD scores for ATAC-seq peaks and ASoC annotations can be found in <code>/project2/xinhe/kevinluo/ldsc/annot/ldscores/</code>.</p>
</div>
</div>
<div id="partition-heritability-using-s-ldsc" class="section level2">
<h2>Partition heritability using S-LDSC</h2>
<p><a href="https://github.com/bulik/ldsc/wiki/Partitioned-Heritability" class="uri">https://github.com/bulik/ldsc/wiki/Partitioned-Heritability</a></p>
<div id="prepare-gwas-summary-statistics-in-ldsc-format" class="section level3">
<h3>Prepare GWAS summary statistics in LDSC format</h3>
<p>Convert GWAS summary statistics to the .sumstats format using <code>munge_sumstats.py</code></p>
<p>See this page for <a href="process_GWAS_sumstats_neuro_traits.html">details</a></p>
<p>Converted GWAS summary statistics (LDSC format) are available in <code>/project2/xinhe/kevinluo/GWAS/GWAS_summary_stats/GWAS_from_Min/ldsc_format/</code></p>
</div>
<div id="partition-heritability" class="section level3">
<h3>Partition heritability</h3>
<p>The following <a href="https://github.com/kevinlkx/analysis_pipelines/tree/master/code/sldsc_annot_baselineLD_separate.sbatch">code</a> estimates the partitioned heritability and enrichment for annotations</p>
<pre class="bash"><code>#!/bin/bash

#SBATCH --job-name=sldsc
#SBATCH --output=sldsc_%J.out
#SBATCH --error=sldsc_%J.err
#SBATCH --partition=broadwl
#SBATCH --mem=10G

dir_GWAS=$1
trait=$2
prefix_annot=$3
dir_sLDSC_output=$4

dir_LDSC=/project2/xinhe/kevinluo/ldsc
dir_ldsc_annot=/project2/xinhe/kevinluo/ldsc/annot/ldscores
dir_baselineLD=/project2/xinhe/kevinluo/ldsc/LDSCORE/1000G_Phase3_baselineLD_v1.1_ldscores

conda activate ldsc

echo &quot;GWAS trait: ${trait}&quot;

dir_out=${dir_sLDSC_output}/${trait}/baselineLDv1.1
mkdir -p ${dir_out}

python $HOME/softwares/ldsc/ldsc.py \
--h2 ${dir_GWAS}/${trait}.sumstats.gz \
--ref-ld-chr ${dir_baselineLD}/baselineLD.,${dir_ldsc_annot}/${prefix_annot}/${prefix_annot}. \
--frqfile-chr ${dir_LDSC}/LDSCORE/1000G_Phase3_frq/1000G.EUR.QC. \
--w-ld-chr ${dir_LDSC}/LDSCORE/1000G_Phase3_weights_hm3_no_MHC/weights.hm3_noMHC. \
--overlap-annot --print-cov --print-coefficients --print-delete-vals \
--out ${dir_out}/${trait}_${prefix_annot}_baselineLDv1.1
</code></pre>
<p>Run S-LDSC across a number of GWAS traits over the ATAC-seq peaks and ASoC annotations.</p>
<p>Results are saved in <code>/project2/xinhe/kevinluo/ldsc/results/sLDSC_neuron_ATACseq_examples/</code></p>
<pre class="bash"><code>
TRAITS=(&quot;ADHD&quot; &quot;IBD&quot; &quot;BMI&quot; &quot;height&quot; &quot;SCZ&quot; &quot;BIP&quot; &quot;MDD&quot; &quot;iPSYCH_ASD&quot; &quot;Intelligence&quot; &quot;Education&quot; &quot;Neuroticism&quot; &quot;Alzheimer&quot; &quot;Parkinson&quot;)
dir_GWAS=/project2/xinhe/kevinluo/GWAS/GWAS_summary_stats/GWAS_from_Min/ldsc_format/
dir_sLDSC_output=/project2/xinhe/kevinluo/ldsc/results/sLDSC_neuron_ATACseq_examples/
dir_code=~/projects/analysis_pipelines/code/

for trait in &quot;${TRAITS[@]}&quot;
do
  sbatch ${dir_code}/sldsc_annot_baselineLD_separate.sbatch ${dir_GWAS} ${trait} CN_all_peaks.narrowPeak.cleaned.hg19.merged ${dir_sLDSC_output}

  sbatch ${dir_code}/sldsc_annot_baselineLD_separate.sbatch ${dir_GWAS} ${trait} DN_all_peaks.narrowPeak.cleaned.hg19.merged ${dir_sLDSC_output}

  sbatch ${dir_code}/sldsc_annot_baselineLD_separate.sbatch ${dir_GWAS} ${trait} GA_all_peaks.narrowPeak.cleaned.hg19.merged ${dir_sLDSC_output}

  sbatch ${dir_code}/sldsc_annot_baselineLD_separate.sbatch ${dir_GWAS} ${trait} ips_all_peaks.narrowPeak.cleaned.hg19.merged ${dir_sLDSC_output}

  sbatch ${dir_code}/sldsc_annot_baselineLD_separate.sbatch ${dir_GWAS} ${trait} NSC_all_peaks.narrowPeak.cleaned.hg19.merged ${dir_sLDSC_output}

  sbatch ${dir_code}/sldsc_annot_baselineLD_separate.sbatch ${dir_GWAS} ${trait} ASoC_glut_anno_hg19 ${dir_sLDSC_output}

  sbatch ${dir_code}/sldsc_annot_baselineLD_separate.sbatch ${dir_GWAS} ${trait} ASoC_npc_anno_hg19 ${dir_sLDSC_output}
done
</code></pre>
</div>
</div>
<div id="extract-and-plot-s-ldsc-enrichment-results" class="section level2">
<h2>Extract and plot S-LDSC enrichment results</h2>
<pre class="r"><code>annot_list &lt;- c(&quot;iN-Glut ASoC&quot;, &quot;NPC ASoC&quot;, &quot;iN-GA OCR&quot;, &quot;iN-DN OCR&quot;, &quot;NPC OCR&quot;, &quot;iN-Glut OCR&quot;, &quot;iPSC OCR&quot;)
trait_name_list &lt;- c(&quot;SCZ&quot;, &quot;BIP&quot;, &quot;MDD&quot;, &quot;Intelligence&quot;, &quot;IBD&quot;)</code></pre>
<pre class="r"><code>library(ggplot2)
library(foreach)
library(doParallel)</code></pre>
<pre><code>Loading required package: iterators</code></pre>
<pre><code>Loading required package: parallel</code></pre>
<pre class="r"><code>## Plot enrichment
ggplot_enrichment &lt;- function(result_sLDSC, xlim = NULL, title = &quot;Enrichment&quot;){
  
  Enrichment &lt;- result_sLDSC$Enrichment
  Enrichment_CI_L &lt;- result_sLDSC$Enrichment - 1.96*result_sLDSC$Enrichment_std_error
  Enrichment_CI_H &lt;- result_sLDSC$Enrichment + 1.96*result_sLDSC$Enrichment_std_error

  ## truncate at 1
  Enrichment[Enrichment &lt; 1] &lt;- 1
  Enrichment_CI_L[Enrichment_CI_L &lt; 1] &lt;- 1
  Enrichment_CI_H[Enrichment_CI_H &lt; 1] &lt;- 1
  
  p &lt;- ggplot(result_sLDSC, aes(x = Enrichment, y = Category, colour = Color))+
    geom_point()+
    xlab(&quot;Enrichment&quot;)+
    ggtitle(title)+
    geom_errorbarh(aes(xmin = Enrichment - 1.96*Enrichment_std_error, 
                       xmax = Enrichment + 1.96*Enrichment_std_error, height = 0.1))+ 
    facet_wrap(Disease~.,ncol = 3)+
    theme_bw()  + 
    geom_vline(xintercept = 1,linetype=&quot;dotted&quot;, colour = &quot;red&quot;)+
    theme(axis.ticks = element_blank(),  
          panel.grid.minor = element_blank(), 
          axis.line = element_line(colour = &quot;black&quot;), 
          axis.text = element_text(face=&quot;bold&quot;,size = 12, colour = &quot;black&quot;),
          axis.title = element_text(face=&quot;bold&quot;,size = 12),
          strip.text = element_text(face=&quot;bold&quot;,size = 12), 
          panel.spacing.x = unit(0.6,units = &quot;cm&quot;), 
          axis.title.y = element_blank(), 
          legend.position = &quot;none&quot;, 
          plot.title = element_text(hjust = 0.5))
  if(!is.null(xlim)){
    p &lt;- p + coord_cartesian(xlim = xlim)
  }
  print(p)
}

ggplot_log2_enrichment &lt;- function(result_sLDSC, xlim = NULL, title = &quot;Enrichment&quot;){
  
  result_sLDSC$Enrichment_CI_L &lt;- result_sLDSC$Enrichment - 1.96*result_sLDSC$Enrichment_std_error
  result_sLDSC$Enrichment_CI_H &lt;- result_sLDSC$Enrichment + 1.96*result_sLDSC$Enrichment_std_error

  ## truncate at 1
  result_sLDSC$Enrichment[result_sLDSC$Enrichment &lt; 1] &lt;- 1
  result_sLDSC$Enrichment_CI_L[result_sLDSC$Enrichment_CI_L &lt; 1] &lt;- 1
  result_sLDSC$Enrichment_CI_H[result_sLDSC$Enrichment_CI_H &lt; 1] &lt;- 1

  p &lt;- ggplot(result_sLDSC, aes(x = log2(Enrichment), y = Category, colour = Color))+
    geom_point()+
    xlab(&quot;log2(Enrichment)&quot;)+
    ggtitle(title)+
    geom_errorbarh(aes(xmin = log2(Enrichment_CI_L), 
                       xmax = log2(Enrichment_CI_H), height = 0.1))+ 
    facet_wrap(Disease~.,ncol = 3)+
    theme_bw()  + 
    geom_vline(xintercept = 0,linetype=&quot;dotted&quot;, colour = &quot;red&quot;)+
    theme(axis.ticks = element_blank(),  
          panel.grid.minor = element_blank(), 
          axis.line = element_line(colour = &quot;black&quot;), 
          axis.text = element_text(face=&quot;bold&quot;,size = 12, colour = &quot;black&quot;),
          axis.title = element_text(face=&quot;bold&quot;,size = 12),
          strip.text = element_text(face=&quot;bold&quot;,size = 12), 
          panel.spacing.x = unit(0.6,units = &quot;cm&quot;), 
          axis.title.y = element_blank(), 
          legend.position = &quot;none&quot;, 
          plot.title = element_text(hjust = 0.5))
  if(!is.null(xlim)){
    p &lt;- p + coord_cartesian(xlim = xlim)
  }
  print(p)
}

ggplot_heritability &lt;- function(result_sLDSC, xlim = NULL, title = &quot;Heritability&quot;){
  ## Proportion of heritability
  p &lt;- ggplot(result_sLDSC, aes(x = Prop._h2*100, y = Category, colour = Color))+
    geom_point()+
    xlab(&quot;Heritability %&quot;)+
    ggtitle(title)+
    geom_errorbarh(aes(xmin = (Prop._h2-1.96*Prop._h2_std_error)*100, 
                       xmax = (Prop._h2+1.96*Prop._h2_std_error)*100, height = 0.1))+ 
    facet_wrap(Disease~.,ncol = 3)+
    theme_bw()  + 
    geom_vline(xintercept = 0,linetype=&quot;dotted&quot;, colour = &quot;red&quot;)+
    theme(axis.ticks = element_blank(),  
          panel.grid.minor = element_blank(), 
          axis.line = element_line(colour = &quot;black&quot;), 
          axis.text = element_text(face=&quot;bold&quot;,size = 12, colour = &quot;black&quot;),
          axis.title = element_text(face=&quot;bold&quot;,size = 12),
          strip.text = element_text(face=&quot;bold&quot;,size = 12), 
          panel.spacing.x = unit(0.6,units = &quot;cm&quot;), 
          axis.title.y = element_blank(), 
          legend.position = &quot;none&quot;, 
          plot.title = element_text(hjust = 0.5))
  if(!is.null(xlim)){
    p &lt;- p + coord_cartesian(xlim = xlim)
  }
  print(p)
}

## change names for traits
change_annot_names &lt;- function(annot_list){
  annot_list &lt;- gsub(&quot;^CN$&quot;,&quot;iN-Glut OCR&quot;, annot_list)
  annot_list &lt;- gsub(&quot;^DN$&quot;,&quot;iN-DN OCR&quot;, annot_list)
  annot_list &lt;- gsub(&quot;^GA$&quot;,&quot;iN-GA OCR&quot;, annot_list)
  annot_list &lt;- gsub(&quot;^ips$&quot;,&quot;iPSC OCR&quot;, annot_list)
  annot_list &lt;- gsub(&quot;^NSC$&quot;,&quot;NPC OCR&quot;, annot_list)
  annot_list &lt;- gsub(&quot;^ASoC_glut$&quot;,&quot;iN-Glut ASoC&quot;, annot_list)
  annot_list &lt;- gsub(&quot;^ASoC_npc$&quot;,&quot;NPC ASoC&quot;, annot_list)
  return(annot_list)
}

## combine S-LDSC enrichment results across traits
combine_sldsc_traits &lt;- function(trait_name_list, dir_results, baseline){
  registerDoParallel(cores = 10)
  
  result_sLDSC &lt;- foreach(trait = trait_name_list, .combine = rbind)%dopar%{
    sldsc_results &lt;- read.table(paste0(dir_results,&quot;/&quot;, trait, &quot;/&quot;, baseline, &quot;/&quot;, trait,&quot;_&quot;, &quot;CN_all_peaks.narrowPeak.cleaned.hg19.merged&quot;, &quot;_&quot;, baseline, &quot;.results&quot;), header = T, stringsAsFactors = F)
    sldsc.CN &lt;- sldsc_results[sldsc_results$Category == &quot;L2_1&quot;,]
    sldsc.CN$Category &lt;- &quot;CN&quot;
    
    sldsc_results &lt;- read.table(paste0(dir_results,&quot;/&quot;, trait, &quot;/&quot;, baseline, &quot;/&quot;, trait,&quot;_&quot;, &quot;DN_all_peaks.narrowPeak.cleaned.hg19.merged&quot;, &quot;_&quot;, baseline, &quot;.results&quot;), header = T, stringsAsFactors = F)
    sldsc.DN &lt;- sldsc_results[sldsc_results$Category == &quot;L2_1&quot;,]
    sldsc.DN$Category &lt;- &quot;DN&quot;
    
    sldsc_results &lt;- read.table(paste0(dir_results,&quot;/&quot;, trait, &quot;/&quot;, baseline, &quot;/&quot;, trait,&quot;_&quot;, &quot;GA_all_peaks.narrowPeak.cleaned.hg19.merged&quot;, &quot;_&quot;, baseline, &quot;.results&quot;), header = T, stringsAsFactors = F)
    sldsc.GA &lt;- sldsc_results[sldsc_results$Category == &quot;L2_1&quot;,]
    sldsc.GA$Category &lt;- &quot;GA&quot;
    
    sldsc_results &lt;- read.table(paste0(dir_results,&quot;/&quot;, trait, &quot;/&quot;, baseline, &quot;/&quot;, trait,&quot;_&quot;, &quot;ips_all_peaks.narrowPeak.cleaned.hg19.merged&quot;, &quot;_&quot;, baseline, &quot;.results&quot;), header = T, stringsAsFactors = F)
    sldsc.ips &lt;- sldsc_results[sldsc_results$Category == &quot;L2_1&quot;,]
    sldsc.ips$Category &lt;- &quot;ips&quot;
    
    sldsc_results &lt;- read.table(paste0(dir_results,&quot;/&quot;, trait, &quot;/&quot;, baseline, &quot;/&quot;, trait,&quot;_&quot;, &quot;NSC_all_peaks.narrowPeak.cleaned.hg19.merged&quot;, &quot;_&quot;, baseline, &quot;.results&quot;), header = T, stringsAsFactors = F)
    sldsc.NSC &lt;- sldsc_results[sldsc_results$Category == &quot;L2_1&quot;,]
    sldsc.NSC$Category &lt;- &quot;NSC&quot;
    
    sldsc_results &lt;- read.table(paste0(dir_results,&quot;/&quot;, trait, &quot;/&quot;, baseline, &quot;/&quot;, trait,&quot;_&quot;, &quot;ASoC_glut_anno_hg19&quot;, &quot;_&quot;, baseline, &quot;.results&quot;), header = T, stringsAsFactors = F)
    sldsc.ASoC_glut &lt;- sldsc_results[sldsc_results$Category == &quot;L2_1&quot;,]
    sldsc.ASoC_glut$Category &lt;- &quot;ASoC_glut&quot;
    
    sldsc_results &lt;- read.table(paste0(dir_results,&quot;/&quot;, trait, &quot;/&quot;, baseline, &quot;/&quot;, trait,&quot;_&quot;, &quot;ASoC_npc_anno_hg19&quot;, &quot;_&quot;, baseline, &quot;.results&quot;), header = T, stringsAsFactors = F)
    sldsc.ASoC_npc &lt;- sldsc_results[sldsc_results$Category == &quot;L2_1&quot;,]
    sldsc.ASoC_npc$Category &lt;- &quot;ASoC_npc&quot;
    
    sldsc.combined &lt;- rbind(sldsc.CN, sldsc.DN, sldsc.GA, sldsc.ips, sldsc.NSC, sldsc.ASoC_glut, sldsc.ASoC_npc)
    sldsc.combined &lt;- cbind(Disease = trait, sldsc.combined)
    sldsc.combined
  }
  return(result_sLDSC)
}</code></pre>
<pre class="r"><code>baseline &lt;- &quot;baselineLDv1.1&quot;

dir_results &lt;- &quot;/project2/xinhe/kevinluo/ldsc/results/sLDSC_neuron_ATACseq_examples/&quot;
result_sLDSC &lt;- combine_sldsc_traits(trait_name_list, dir_results, baseline)

result_sLDSC$Category &lt;- change_annot_names(result_sLDSC$Category)

result_sLDSC$Category &lt;- factor(result_sLDSC$Category, levels = rev(annot_list) )
result_sLDSC$Color &lt;- factor(result_sLDSC$Category, levels = annot_list)</code></pre>
<div id="enrichment" class="section level4">
<h4>Enrichment</h4>
<pre class="r"><code>DT::datatable(format(result_sLDSC[,1:7], digits = 2), options = list(scrollX = TRUE, keys = TRUE, pageLength = length(annot_list)),rownames = F)</code></pre>
<div id="htmlwidget-8316daeedf19885da48f" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-8316daeedf19885da48f">{"x":{"filter":"none","data":[["SCZ","SCZ","SCZ","SCZ","SCZ","SCZ","SCZ","BIP","BIP","BIP","BIP","BIP","BIP","BIP","MDD","MDD","MDD","MDD","MDD","MDD","MDD","Intelligence","Intelligence","Intelligence","Intelligence","Intelligence","Intelligence","Intelligence","IBD","IBD","IBD","IBD","IBD","IBD","IBD"],["iN-Glut OCR","iN-DN OCR","iN-GA OCR","iPSC OCR","NPC OCR","iN-Glut ASoC","NPC ASoC","iN-Glut OCR","iN-DN OCR","iN-GA OCR","iPSC OCR","NPC OCR","iN-Glut ASoC","NPC ASoC","iN-Glut OCR","iN-DN OCR","iN-GA OCR","iPSC OCR","NPC OCR","iN-Glut ASoC","NPC ASoC","iN-Glut OCR","iN-DN OCR","iN-GA OCR","iPSC OCR","NPC OCR","iN-Glut ASoC","NPC ASoC","iN-Glut OCR","iN-DN OCR","iN-GA OCR","iPSC OCR","NPC OCR","iN-Glut ASoC","NPC ASoC"],["0.03635","0.05724","0.05900","0.08299","0.03188","0.00079","0.00050","0.03635","0.05724","0.05900","0.08299","0.03188","0.00079","0.00050","0.03635","0.05724","0.05900","0.08299","0.03188","0.00079","0.00050","0.03635","0.05724","0.05900","0.08299","0.03188","0.00079","0.00050","0.03635","0.05724","0.05900","0.08299","0.03188","0.00079","0.00050"],[" 0.1496"," 0.2280"," 0.2616"," 0.1649"," 0.1285"," 0.0109"," 0.0091"," 0.1267"," 0.2396"," 0.2538"," 0.1396"," 0.1056"," 0.0112"," 0.0198"," 0.1204"," 0.2308"," 0.3042"," 0.0804"," 0.0954"," 0.0181"," 0.0102"," 0.2022"," 0.3174"," 0.2871"," 0.2473"," 0.1776"," 0.0147"," 0.0069","-0.1578","-0.1867","-0.2022","-0.1876","-0.0888"," 0.0033","-0.0054"],["0.0298","0.0335","0.0331","0.0402","0.0295","0.0073","0.0049","0.0487","0.0544","0.0561","0.0623","0.0423","0.0118","0.0086","0.0562","0.0664","0.0654","0.0718","0.0537","0.0114","0.0096","0.0246","0.0321","0.0310","0.0337","0.0255","0.0059","0.0055","0.0740","0.0960","0.0872","0.1168","0.0743","0.0148","0.0101"],["  4.11","  3.98","  4.43","  1.99","  4.03"," 13.77"," 18.37","  3.48","  4.19","  4.30","  1.68","  3.31"," 14.16"," 39.85","  3.31","  4.03","  5.16","  0.97","  2.99"," 22.84"," 20.63","  5.56","  5.54","  4.87","  2.98","  5.57"," 18.62"," 13.91"," -4.34"," -3.26"," -3.43"," -2.26"," -2.79","  4.19","-10.89"],[" 0.82"," 0.59"," 0.56"," 0.48"," 0.92"," 9.22"," 9.97"," 1.34"," 0.95"," 0.95"," 0.75"," 1.33","14.92","17.35"," 1.55"," 1.16"," 1.11"," 0.86"," 1.68","14.44","19.29"," 0.68"," 0.56"," 0.53"," 0.41"," 0.80"," 7.50","11.10"," 2.03"," 1.68"," 1.48"," 1.41"," 2.33","18.74","20.42"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>Disease<\/th>\n      <th>Category<\/th>\n      <th>Prop._SNPs<\/th>\n      <th>Prop._h2<\/th>\n      <th>Prop._h2_std_error<\/th>\n      <th>Enrichment<\/th>\n      <th>Enrichment_std_error<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"scrollX":true,"keys":true,"pageLength":7,"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[7,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>ggplot_enrichment(result_sLDSC, title = &quot;&quot;, xlim = c(0,50))</code></pre>
<p><img src="figure/sldsc_example_neuron_ATACseq.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-1">
Past versions of unnamed-chunk-10-1.png
</button>
</p>
<div id="fig-unnamed-chunk-10-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/kevinlkx/analysis_pipelines/blob/03346651e2081f52eb9fe4bdf16d120d3233180b/docs/figure/sldsc_example_neuron_ATACseq.Rmd/unnamed-chunk-10-1.png" target="_blank">0334665</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-07-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="log2-enrichment" class="section level4">
<h4>log2 Enrichment</h4>
<pre class="r"><code>ggplot_log2_enrichment(result_sLDSC, title = &quot;&quot;)</code></pre>
<p><img src="figure/sldsc_example_neuron_ATACseq.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-1">
Past versions of unnamed-chunk-11-1.png
</button>
</p>
<div id="fig-unnamed-chunk-11-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/kevinlkx/analysis_pipelines/blob/03346651e2081f52eb9fe4bdf16d120d3233180b/docs/figure/sldsc_example_neuron_ATACseq.Rmd/unnamed-chunk-11-1.png" target="_blank">0334665</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-07-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.1 (2018-07-02)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] parallel  stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
[1] doParallel_1.0.14 iterators_1.0.12  foreach_1.5.0     ggplot2_3.3.0    
[5] workflowr_1.6.2  

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.4.6      compiler_3.5.1    pillar_1.4.4      later_1.0.0      
 [5] git2r_0.27.1      tools_3.5.1       digest_0.6.25     jsonlite_1.6     
 [9] evaluate_0.14     lifecycle_0.2.0   tibble_3.0.1      gtable_0.3.0     
[13] pkgconfig_2.0.3   rlang_0.4.6       shiny_1.4.0.2     crosstalk_1.0.0  
[17] yaml_2.2.0        xfun_0.14         fastmap_1.0.1     withr_2.1.2      
[21] stringr_1.4.0     dplyr_0.8.5       knitr_1.28        htmlwidgets_1.5.1
[25] fs_1.3.1          vctrs_0.3.0       DT_0.13           rprojroot_1.3-2  
[29] grid_3.5.1        tidyselect_0.2.5  glue_1.4.1        R6_2.4.1         
[33] rmarkdown_2.1     farver_2.0.3      purrr_0.3.4       magrittr_1.5     
[37] whisker_0.4       codetools_0.2-15  backports_1.1.7   scales_1.1.1     
[41] promises_1.1.0    htmltools_0.4.0   ellipsis_0.3.1    assertthat_0.2.1 
[45] xtable_1.8-4      mime_0.9          colorspace_1.4-1  httpuv_1.5.3.1   
[49] labeling_0.3      stringi_1.4.6     munsell_0.5.0     crayon_1.3.4     </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
