<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Kaixuan Luo" />

<meta name="date" content="2020-05-24" />

<title>S-LSDC pipeline</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Analysis pipelines</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">S-LSDC pipeline</h1>
<h4 class="author">Kaixuan Luo</h4>
<h4 class="date">2020-05-24</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-05-24
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>analysis_pipelines/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200524code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200524)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200524code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200524)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongd07ad1d"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> d07ad1d </a>
</p>
</div>
<div id="strongRepositoryversionstrongd07ad1d" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Untracked files:
    Untracked:  analysis/GWAS_sumstats.Rmd
    Untracked:  analysis/TWAS.Rmd
    Untracked:  analysis/ldsc_setup.Rmd
    Untracked:  code/make_ldsc_binary_annot.R
    Untracked:  code/make_ldsc_continuous_annot.R

Unstaged changes:
    Modified:   analysis/_site.yml
    Modified:   analysis/index.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
d07ad1d
</td>
<td>
kevinlkx
</td>
<td>
2020-05-24
</td>
<td>
wflow_publish(“analysis/sldsc_pipeline.Rmd”)
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="setup" class="section level2">
<h2>Setup</h2>
<p>Please see the detail instructions: LD Score Regression (LDSC) <a href="https://github.com/bulik/ldsc" class="uri">https://github.com/bulik/ldsc</a></p>
<div id="activate-the-conda-environment-with-ldscs-dependencies" class="section level3">
<h3>Activate the conda environment with LDSC’s dependencies</h3>
<pre class="bash"><code>conda activate ldsc</code></pre>
</div>
</div>
<div id="tutorials" class="section level2">
<h2>Tutorials</h2>
<ul>
<li><p><a href="https://github.com/bulik/ldsc/wiki/Partitioned-Heritability">Partitioned Heritability</a></p></li>
<li><p><a href="https://github.com/bulik/ldsc/wiki/Partitioned-Heritability-from-Continuous-Annotations">Partitioned Heritability from Continuous Annotations</a></p></li>
</ul>
</div>
<div id="download-data" class="section level2">
<h2>Download data</h2>
<p>You may need to download some of the following datasets:</p>
<p>Most of the data can be downloaded from the Price lab <a href="https://data.broadinstitute.org/alkesgroup/LDSCORE/">LDSCORE</a> website</p>
<div id="download-the-baseline-model-ld-scores" class="section level4">
<h4>Download the baseline model LD scores</h4>
<ul>
<li><p><a href="https://data.broadinstitute.org/alkesgroup/LDSCORE/readme_baseline_versions">Readme</a> of different versions of baseline models</p></li>
<li><p>1000G Phase3 baseline model v1.1</p></li>
</ul>
<pre class="bash"><code>wget https://data.broadinstitute.org/alkesgroup/LDSCORE/1000G_Phase3_baseline_v1.1_ldscores.tgz
mkdir 1000G_Phase3_baseline_v1.1_ldscores
tar -xvzf 1000G_Phase3_baseline_v1.1_ldscores.tgz -C 1000G_Phase3_baseline_v1.1_ldscores</code></pre>
<ul>
<li>1000G Phase3 baselineLD model v1.1</li>
</ul>
<pre class="bash"><code>wget https://data.broadinstitute.org/alkesgroup/LDSCORE/1000G_Phase3_baselineLD_v1.1_ldscores.tgz
mkdir 1000G_Phase3_baselineLD_v1.1_ldscores
tar -xvzf 1000G_Phase3_baselineLD_v1.1_ldscores.tgz -C 1000G_Phase3_baselineLD_v1.1_ldscores</code></pre>
<ul>
<li>1000G Phase3 baselineLD model v2.2</li>
</ul>
<pre class="bash"><code>wget https://data.broadinstitute.org/alkesgroup/LDSCORE/1000G_Phase3_baselineLD_v2.2_ldscores.tgz
mkdir 1000G_Phase3_baselineLD_v2.2_ldscores
tar -xvzf 1000G_Phase3_baselineLD_v2.2_ldscores.tgz -C 1000G_Phase3_baselineLD_v2.2_ldscores</code></pre>
</div>
<div id="download-regression-weights" class="section level4">
<h4>Download regression weights</h4>
<pre class="bash"><code># wget https://data.broadinstitute.org/alkesgroup/LDSCORE/weights_hm3_no_hla.tgz
# tar -xvzf weights_hm3_no_hla.tgz

# European of Phase 3 of 1000 Genomes
wget https://data.broadinstitute.org/alkesgroup/LDSCORE/1000G_Phase3_weights_hm3_no_MHC.tgz
tar -xvzf 1000G_Phase3_weights_hm3_no_MHC.tgz</code></pre>
</div>
<div id="download-allele-frequencies-european-of-phase-3-of-1000-genomes" class="section level4">
<h4>Download allele frequencies (European of Phase 3 of 1000 Genomes)</h4>
<pre class="bash"><code>wget https://data.broadinstitute.org/alkesgroup/LDSCORE/1000G_Phase3_frq.tgz
tar -xvzf 1000G_Phase3_frq.tgz</code></pre>
<p>The authors recommend only keeping HapMap3 SNPs. You can download the HapMap3 related files:</p>
</div>
<div id="download-1000-genomes-reference-genotypes-at-hapmap3-loci" class="section level4">
<h4>Download 1000 genomes reference genotypes at HapMap3 loci</h4>
<pre class="bash"><code>wget https://data.broadinstitute.org/alkesgroup/LDSCORE/1000G_Phase3_plinkfiles.tgz
tar -xvzf 1000G_Phase3_plinkfiles.tgz</code></pre>
</div>
<div id="download-hapmap3-snps" class="section level4">
<h4>Download HapMap3 SNPs</h4>
<pre class="bash"><code>wget https://data.broadinstitute.org/alkesgroup/LDSCORE/hapmap3_snps.tgz
tar -xvzf hapmap3_snps.tgz</code></pre>
</div>
<div id="download-a-concatenated-list-of-hapmap3-snps" class="section level4">
<h4>Download a concatenated list of HapMap3 SNPs</h4>
<pre class="bash"><code>wget https://data.broadinstitute.org/alkesgroup/LDSCORE/w_hm3.snplist.bz2
bzip2 -d w_hm3.snplist.bz2

# Extract the list of HapMap 3 SNPs rsIDs
awk &#39;{if ($1!=&quot;SNP&quot;) {print $1} }&#39; w_hm3.snplist &gt; listHM3.txt
</code></pre>
</div>
</div>
<div id="compute-ld-score-from-annotation-files" class="section level2">
<h2>Compute LD score from annotation files</h2>
<ul>
<li><code>ldsc</code> wiki <a href="https://github.com/bulik/ldsc/wiki/LD-Score-Estimation-Tutorial">“LD-Score-Estimation-Tutorial”</a></li>
</ul>
<div id="compute-ld-scores-for-binary-annotations" class="section level3">
<h3>Compute LD scores for binary annotations</h3>
<ul>
<li>Step 1: prepare annotation in UCSC <a href="https://genome.ucsc.edu/FAQ/FAQformat.html#format1">BED</a> format</li>
<li><p>Step 2: compute LD scores using annotation BED files</p></li>
<li><p>Example scripts: <code>thin-annot</code> annotation format using <code>make_annot.py</code></p></li>
</ul>
<pre class="bash"><code>for chrom in {1..22}
do
  echo ${chrom}

  ## Step 1: Creating an annot file
  echo &quot;Make ldsc-friendly annotation files for ${ANNOT}.bed&quot;
  python make_annot.py \
  --bed-file ${ANNOT}.bed \
  --bimfile 1000G_EUR_Phase3_plink/1000G.EUR.QC.${chrom}.bim \
  --annot-file ${ANNOT}.${chrom}.annot.gz

  ## Step 2: Computing LD scores with an annot file
  echo &quot;Computing LD scores with the annot file ${ANNOT}.${chrom}.annot.gz&quot;
  python ldsc.py \
  --l2 \
  --bfile 1000G_EUR_Phase3_plink/1000G.EUR.QC.${chrom} \
  --print-snps listHM3.txt \
  --ld-wind-cm 1 \
  --annot ${ANNOT}.${chrom}.annot.gz \
  --thin-annot \
  --out ${ANNOT}.${chrom}
done</code></pre>
<p>full annotation or <code>thin-annot</code> format using my own <a href="../code/make_ldsc_binary_annot.R">code</a></p>
<pre class="bash"><code>for chrom in {1..22}
do
  echo ${chrom}

  ## Step 1: Creating an annot file
  echo &quot;Make ldsc-friendly annotation files for ${ANNOT}.bed&quot;
  Rscript code/make_ldsc_binary_annot.R \
  ${ANNOT}.bed \
  1000G_EUR_Phase3_plink/1000G.EUR.QC.${chrom}.bim \
  ${ANNOT}.${chrom}.annot.gz &quot;full-annot&quot;

  ## Step 2: Computing LD scores with an annot file
  echo &quot;Computing LD scores with the annot file ${ANNOT}.${chrom}.annot.gz&quot;
  python ldsc.py \
  --l2 \
  --bfile 1000G_EUR_Phase3_plink/1000G.EUR.QC.${chrom} \
  --print-snps listHM3.txt \
  --ld-wind-cm 1 \
  --annot ${ANNOT}.${chrom}.annot.gz \
  --out ${ANNOT}.${chrom}
done</code></pre>
</div>
<div id="compute-ld-scores-for-continuous-annotations" class="section level3">
<h3>Compute LD scores for continuous annotations</h3>
<ul>
<li>Step 1: prepare annotation in UCSC <a href="https://genome.ucsc.edu/FAQ/FAQformat.html#format1">BED</a> format</li>
<li><p>Step 2: compute LD scores using annotation BED files</p></li>
<li><p>Example script (full annotation format using my own <a href="../code/make_ldsc_continuous_annot.R">code</a> )</p></li>
</ul>
<pre class="bash"><code>
for chrom in {1..22}
do
  echo ${chrom}

  ## Step 1: Creating an annot file (using make_ldsc_continuous_annot.R)
  echo &quot;Make ldsc-friendly annotation files for ${ANNOT}.bed&quot;
  Rscript code/make_ldsc_continuous_annot.R \
  ${ANNOT}.bed \
  1000G_EUR_Phase3_plink/1000G.EUR.QC.${chrom}.bim \
  ${ANNOT}.${chrom}.annot.gz &quot;full-annot&quot;

  ## Step 2: Computing LD scores with an annot file
  echo &quot;Computing LD scores with the annot file ${ANNOT}.${chrom}.annot.gz&quot;
  python ldsc.py \
  --l2 \
  --bfile 1000G_EUR_Phase3_plink/1000G.EUR.QC.${chrom} \
  --print-snps listHM3.txt \
  --ld-wind-cm 1 \
  --annot ${ANNOT}.${chrom}.annot.gz \
  --out ${ANNOT}.${chrom}

done
</code></pre>
</div>
</div>
<div id="partitioned-heritability-binary-annotation" class="section level2">
<h2>Partitioned heritability (binary annotation)</h2>
<ul>
<li><code>ldsc</code> wiki <a href="https://github.com/bulik/ldsc/wiki/Partitioned-Heritability">“Partitioned-Heritability”</a></li>
</ul>
<div id="prepare-gwas-summary-stats-in-ldsc-.sumstats-format" class="section level3">
<h3>Prepare GWAS summary stats in LDSC <code>.sumstats</code> format</h3>
<ul>
<li>Convert GWAS summary stats to the LDSC <code>.sumstats</code> format using <code>munge_sumstats.py</code></li>
<li><p><code>ldsc</code> wiki <a href="https://github.com/bulik/ldsc/wiki/Partitioned-Heritability">“Summary-Statistics-File-Format”</a></p></li>
<li><p>Some of the processed GWAS summary stats can be found on RCC: <code>/project2/xinhe/kevinluo/GWAS/GWAS_summary_stats/GWAS_collection/ldsc_format/</code></p></li>
</ul>
</div>
<div id="run-ldsc-on-your-gwas-summary-statistics-using-baseline-ld-model-annotations-and-your-new-annotation" class="section level3">
<h3>Run ldsc on your GWAS summary statistics using baseline-LD model annotations and your new annotation</h3>
<ul>
<li>Compute the annotation conditional on baselineLD model: controlling for the annotation categories of the full baselineLD model, using a comma-separated list of annotation file prefixes with <code>--ref-ld-chr</code></li>
</ul>
<pre class="bash"><code>python ldsc.py \
--h2 ${TRAIT}.sumstats.gz \
--ref-ld-chr baselineLD.,${ANNOT}. \
--frqfile-chr 1000G_Phase3_frq/1000G.EUR.QC. \
--w-ld-chr 1000G_Phase3_weights_hm3_no_MHC/weights.hm3_noMHC. \
--overlap-annot --print-cov --print-coefficients --print-delete-vals \
--out ${TRAIT}_${ANNOT}_baselineLD</code></pre>
<ul>
<li>Joint model: you could include multiple annotations file prefixes to run multiple annotations jointly</li>
</ul>
<pre class="bash"><code>python ldsc.py \
--h2 ${TRAIT}.sumstats.gz \
--ref-ld-chr baselineLD.,${ANNOT_1}.,${ANNOT_2}. \
--frqfile-chr 1000G_Phase3_frq/1000G.EUR.QC. \
--w-ld-chr 1000G_Phase3_weights_hm3_no_MHC/weights.hm3_noMHC. \
--overlap-annot --print-cov --print-coefficients --print-delete-vals \
--out ${TRAIT}_joint_baselineLD</code></pre>
</div>
</div>
<div id="partitioned-heritability-continuous-annotation" class="section level2">
<h2>Partitioned heritability (continuous annotation)</h2>
<ul>
<li><p><code>ldsc</code> allows taking continuous annotations as inputs for both –l2 and –h2 options. The pipeline is similar to that using binary annotation. However, some result outputs of –h2 option are not meaningful anymore. Computing the proportion of heritability explained by each quantile of a continuous annotation provides a more intuitive interpretation of the magnitude of a continuous annotation effects. You can use their R script <code>quantile_h2g.r</code> and follow their wiki tutorial to compute the proportion of heritability explained by each quintile.</p></li>
<li><p>Please follow the <code>ldsc</code> wiki <a href="https://github.com/bulik/ldsc/wiki/Partitioned-Heritability-from-Continuous-Annotations">“Partitioned Heritability from Continuous Annotations”</a></p></li>
</ul>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.1 (2018-07-02)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] workflowr_1.6.0

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.3        rprojroot_1.3-2   digest_0.6.23     later_1.0.0      
 [5] R6_2.4.1          backports_1.1.5   git2r_0.26.1.9000 magrittr_1.5     
 [9] evaluate_0.14     stringi_1.4.5     rlang_0.4.4       fs_1.3.1         
[13] promises_1.1.0    whisker_0.4       rmarkdown_2.1     tools_3.5.1      
[17] stringr_1.4.0     glue_1.3.1        httpuv_1.5.2      xfun_0.12        
[21] yaml_2.2.0        compiler_3.5.1    htmltools_0.4.0   knitr_1.28       </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
