---
title: "S-LDSC example analysis to estimate the enrichment of GWAS variants in neuron ATAC-seq OCR and ASoC variants"
author: "Kaixuan Luo"
date: "07/08/2020"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

This tutorial shows an example analysis using S-LDSC analysis to estimate the enrichment of GWAS variants in neuron ATAC-seq annotations: open chromatin regions (OCR) and allele-specific open chromatin (ASoC) variants.

## Prepare annotation files and compute LD scores for annotations

### Prepare annotations in BED format for ASoC binary annotations
* Convert ASoC binary annotations to BED format. The annotations can be found in `/project2/xinhe/kevinluo/ldsc/annot/annot_bed/`
* Annotations for ATAC-seq peaks in BED format can also be found in `/project2/xinhe/kevinluo/ldsc/annot/annot_bed/`.
```{r, eval=FALSE}
## Prepare ASoC binary annotations in BED format for LDSC analysis
dir_annot_bed <- "/project2/xinhe/kevinluo/ldsc/annot/annot_bed/"

### ASoC_glut_anno_hg19
annot_name <- "ASoC_glut_anno_hg19"
annot_filename <- "ASoC_glut_anno_hg19.txt"
ASoC_annot <- read.table(paste0(dir_annot_bed, "/", annot_filename), header = F, stringsAsFactors = F)
colnames(ASoC_annot) <- c("chr", "SNP_POS", "annot")

ASoC_sig <- ASoC_annot[ASoC_annot$annot == 1, ]
ASoC_sig.bed <- data.frame(chr = ASoC_sig$chr, start = ASoC_sig$SNP_POS - 1, end = ASoC_sig$SNP_POS)
ASoC_sig.bed$chr <- factor(ASoC_sig.bed$chr, levels = paste0("chr", 1:22))
ASoC_sig.bed <- ASoC_sig.bed[order(ASoC_sig.bed$chr, ASoC_sig.bed$start), ]
ASoC_sig.bed <- unique(ASoC_sig.bed)
cat(nrow(ASoC_sig.bed), "SNPs with annotation:", annot_name, "\n")
write.table(ASoC_sig.bed, paste0(dir_annot_bed, "/", annot_name, ".bed"), sep = "\t", col.names = F, row.names = F, quote = F)

### ASoC_npc_anno_hg19
annot_name <- "ASoC_npc_anno_hg19"
annot_filename <- "ASoC_npc_anno_hg19.txt"
ASoC_annot <- read.table(paste0(dir_annot_bed, "/", annot_filename), header = F, stringsAsFactors = F)
colnames(ASoC_annot) <- c("chr", "SNP_POS", "annot")

ASoC_sig <- ASoC_annot[ASoC_annot$annot == 1, ]
ASoC_sig.bed <- data.frame(chr = ASoC_sig$chr, start = ASoC_sig$SNP_POS - 1, end = ASoC_sig$SNP_POS)
ASoC_sig.bed$chr <- factor(ASoC_sig.bed$chr, levels = paste0("chr", 1:22))
ASoC_sig.bed <- ASoC_sig.bed[order(ASoC_sig.bed$chr, ASoC_sig.bed$start), ]
ASoC_sig.bed <- unique(ASoC_sig.bed)
cat(nrow(ASoC_sig.bed), "SNPs with annotation:", annot_name, "\n")
write.table(ASoC_sig.bed, paste0(dir_annot_bed, "/", annot_name, ".bed"), sep = "\t", col.names = F, row.names = F, quote = F)

```

### Compute LD scores for ATAC-seq peaks and ASoC annotations

The following [code](https://github.com/kevinlkx/analysis_pipelines/tree/master/code/ldsc_binary_annot_QTL.sbatch) generates ldsc-friendly annotation files (annot.gz) from the annotation BED files, then computes LD scores with the annot file (annot.gz).
```{bash, eval=FALSE}
## Compute LD scores for ATAC-seq peak annotations
dir_code=~/projects/analysis_pipelines/code/

sbatch ${dir_code}/ldsc_binary_annot_QTL.sbatch CN_all_peaks.narrowPeak.cleaned.hg19.merged

sbatch ${dir_code}/ldsc_binary_annot_QTL.sbatch DN_all_peaks.narrowPeak.cleaned.hg19.merged

sbatch ${dir_code}/ldsc_binary_annot_QTL.sbatch GA_all_peaks.narrowPeak.cleaned.hg19.merged

sbatch ${dir_code}/ldsc_binary_annot_QTL.sbatch ips_all_peaks.narrowPeak.cleaned.hg19.merged

sbatch ${dir_code}/ldsc_binary_annot_QTL.sbatch NSC_all_peaks.narrowPeak.cleaned.hg19.merged

sbatch ${dir_code}/ldsc_binary_annot_QTL.sbatch NSC_all_peaks.narrowPeak.cleaned.hg19.merged

## Compute LD scores for ASoC annotations
sbatch ${dir_code}/ldsc_binary_annot_QTL.sbatch ASoC_glut_anno_hg19

sbatch ${dir_code}/ldsc_binary_annot_QTL.sbatch ASoC_npc_anno_hg19

```

Computed LD scores for ATAC-seq peaks and ASoC annotations can be found in `/project2/xinhe/kevinluo/ldsc/annot/ldscores/`.

## Partition heritability using S-LDSC
https://github.com/bulik/ldsc/wiki/Partitioned-Heritability

### Prepare GWAS summary statistics in LDSC format
Convert GWAS summary statistics to the .sumstats format using `munge_sumstats.py`

See this page for [details](process_GWAS_sumstats_neuro_traits.html)

Converted GWAS summary statistics (LDSC format) are available in  `/project2/xinhe/kevinluo/GWAS/GWAS_summary_stats/GWAS_from_Min/ldsc_format/`

### Partition heritability

The following [code](https://github.com/kevinlkx/analysis_pipelines/tree/master/code/sldsc_annot_baselineLD_separate.sbatch) estimates the partitioned heritability and enrichment for annotations
```{bash, eval=FALSE}
#!/bin/bash

#SBATCH --job-name=sldsc
#SBATCH --output=sldsc_%J.out
#SBATCH --error=sldsc_%J.err
#SBATCH --partition=broadwl
#SBATCH --mem=10G

dir_GWAS=$1
trait=$2
prefix_annot=$3
dir_sLDSC_output=$4

dir_LDSC=/project2/xinhe/kevinluo/ldsc
dir_ldsc_annot=/project2/xinhe/kevinluo/ldsc/annot/ldscores
dir_baselineLD=/project2/xinhe/kevinluo/ldsc/LDSCORE/1000G_Phase3_baselineLD_v1.1_ldscores

conda activate ldsc

echo "GWAS trait: ${trait}"

dir_out=${dir_sLDSC_output}/${trait}/baselineLDv1.1
mkdir -p ${dir_out}

python $HOME/softwares/ldsc/ldsc.py \
--h2 ${dir_GWAS}/${trait}.sumstats.gz \
--ref-ld-chr ${dir_baselineLD}/baselineLD.,${dir_ldsc_annot}/${prefix_annot}/${prefix_annot}. \
--frqfile-chr ${dir_LDSC}/LDSCORE/1000G_Phase3_frq/1000G.EUR.QC. \
--w-ld-chr ${dir_LDSC}/LDSCORE/1000G_Phase3_weights_hm3_no_MHC/weights.hm3_noMHC. \
--overlap-annot --print-cov --print-coefficients --print-delete-vals \
--out ${dir_out}/${trait}_${prefix_annot}_baselineLDv1.1

```

Run S-LDSC across a number of GWAS traits over the ATAC-seq peaks and ASoC annotations.

Results are saved in `/project2/xinhe/kevinluo/ldsc/results/sLDSC_neuron_ATACseq_examples/`
```{bash, eval=FALSE}

TRAITS=("ADHD" "IBD" "BMI" "height" "SCZ" "BIP" "MDD" "iPSYCH_ASD" "Intelligence" "Education" "Neuroticism" "Alzheimer" "Parkinson")
dir_GWAS=/project2/xinhe/kevinluo/GWAS/GWAS_summary_stats/GWAS_from_Min/ldsc_format/
dir_sLDSC_output=/project2/xinhe/kevinluo/ldsc/results/sLDSC_neuron_ATACseq_examples/
dir_code=~/projects/analysis_pipelines/code/

for trait in "${TRAITS[@]}"
do
  sbatch ${dir_code}/sldsc_annot_baselineLD_separate.sbatch ${dir_GWAS} ${trait} CN_all_peaks.narrowPeak.cleaned.hg19.merged ${dir_sLDSC_output}

  sbatch ${dir_code}/sldsc_annot_baselineLD_separate.sbatch ${dir_GWAS} ${trait} DN_all_peaks.narrowPeak.cleaned.hg19.merged ${dir_sLDSC_output}

  sbatch ${dir_code}/sldsc_annot_baselineLD_separate.sbatch ${dir_GWAS} ${trait} GA_all_peaks.narrowPeak.cleaned.hg19.merged ${dir_sLDSC_output}

  sbatch ${dir_code}/sldsc_annot_baselineLD_separate.sbatch ${dir_GWAS} ${trait} ips_all_peaks.narrowPeak.cleaned.hg19.merged ${dir_sLDSC_output}

  sbatch ${dir_code}/sldsc_annot_baselineLD_separate.sbatch ${dir_GWAS} ${trait} NSC_all_peaks.narrowPeak.cleaned.hg19.merged ${dir_sLDSC_output}

  sbatch ${dir_code}/sldsc_annot_baselineLD_separate.sbatch ${dir_GWAS} ${trait} ASoC_glut_anno_hg19 ${dir_sLDSC_output}

  sbatch ${dir_code}/sldsc_annot_baselineLD_separate.sbatch ${dir_GWAS} ${trait} ASoC_npc_anno_hg19 ${dir_sLDSC_output}
done

```

```{bash, eval=FALSE, include=FALSE}
for trait in "${TRAITS[@]}"
do
  sbatch ${dir_code}/run_sldsc_annots_neuron_ATACseq.sbatch ${dir_GWAS} ${trait} ${dir_sLDSC_output}
done
```

## Extract and plot S-LDSC enrichment results

```{r}
annot_list <- c("iN-Glut ASoC", "NPC ASoC", "iN-GA OCR", "iN-DN OCR", "NPC OCR", "iN-Glut OCR", "iPSC OCR")
trait_name_list <- c("SCZ", "BIP", "MDD", "Intelligence", "IBD")
```

```{r}
library(ggplot2)
library(foreach)
library(doParallel)

## Plot enrichment
ggplot_enrichment <- function(result_sLDSC, xlim = NULL, title = "Enrichment"){
  
  Enrichment <- result_sLDSC$Enrichment
  Enrichment_CI_L <- result_sLDSC$Enrichment - 1.96*result_sLDSC$Enrichment_std_error
  Enrichment_CI_H <- result_sLDSC$Enrichment + 1.96*result_sLDSC$Enrichment_std_error

  ## truncate at 1
  Enrichment[Enrichment < 1] <- 1
  Enrichment_CI_L[Enrichment_CI_L < 1] <- 1
  Enrichment_CI_H[Enrichment_CI_H < 1] <- 1
  
  p <- ggplot(result_sLDSC, aes(x = Enrichment, y = Category, colour = Color))+
    geom_point()+
    xlab("Enrichment")+
    ggtitle(title)+
    geom_errorbarh(aes(xmin = Enrichment - 1.96*Enrichment_std_error, 
                       xmax = Enrichment + 1.96*Enrichment_std_error, height = 0.1))+ 
    facet_wrap(Disease~.,ncol = 3)+
    theme_bw()  + 
    geom_vline(xintercept = 1,linetype="dotted", colour = "red")+
    theme(axis.ticks = element_blank(),  
          panel.grid.minor = element_blank(), 
          axis.line = element_line(colour = "black"), 
          axis.text = element_text(face="bold",size = 12, colour = "black"),
          axis.title = element_text(face="bold",size = 12),
          strip.text = element_text(face="bold",size = 12), 
          panel.spacing.x = unit(0.6,units = "cm"), 
          axis.title.y = element_blank(), 
          legend.position = "none", 
          plot.title = element_text(hjust = 0.5))
  if(!is.null(xlim)){
    p <- p + coord_cartesian(xlim = xlim)
  }
  print(p)
}

ggplot_log2_enrichment <- function(result_sLDSC, xlim = NULL, title = "Enrichment"){
  
  result_sLDSC$Enrichment_CI_L <- result_sLDSC$Enrichment - 1.96*result_sLDSC$Enrichment_std_error
  result_sLDSC$Enrichment_CI_H <- result_sLDSC$Enrichment + 1.96*result_sLDSC$Enrichment_std_error

  ## truncate at 1
  result_sLDSC$Enrichment[result_sLDSC$Enrichment < 1] <- 1
  result_sLDSC$Enrichment_CI_L[result_sLDSC$Enrichment_CI_L < 1] <- 1
  result_sLDSC$Enrichment_CI_H[result_sLDSC$Enrichment_CI_H < 1] <- 1

  p <- ggplot(result_sLDSC, aes(x = log2(Enrichment), y = Category, colour = Color))+
    geom_point()+
    xlab("log2(Enrichment)")+
    ggtitle(title)+
    geom_errorbarh(aes(xmin = log2(Enrichment_CI_L), 
                       xmax = log2(Enrichment_CI_H), height = 0.1))+ 
    facet_wrap(Disease~.,ncol = 3)+
    theme_bw()  + 
    geom_vline(xintercept = 0,linetype="dotted", colour = "red")+
    theme(axis.ticks = element_blank(),  
          panel.grid.minor = element_blank(), 
          axis.line = element_line(colour = "black"), 
          axis.text = element_text(face="bold",size = 12, colour = "black"),
          axis.title = element_text(face="bold",size = 12),
          strip.text = element_text(face="bold",size = 12), 
          panel.spacing.x = unit(0.6,units = "cm"), 
          axis.title.y = element_blank(), 
          legend.position = "none", 
          plot.title = element_text(hjust = 0.5))
  if(!is.null(xlim)){
    p <- p + coord_cartesian(xlim = xlim)
  }
  print(p)
}

ggplot_heritability <- function(result_sLDSC, xlim = NULL, title = "Heritability"){
  ## Proportion of heritability
  p <- ggplot(result_sLDSC, aes(x = Prop._h2*100, y = Category, colour = Color))+
    geom_point()+
    xlab("Heritability %")+
    ggtitle(title)+
    geom_errorbarh(aes(xmin = (Prop._h2-1.96*Prop._h2_std_error)*100, 
                       xmax = (Prop._h2+1.96*Prop._h2_std_error)*100, height = 0.1))+ 
    facet_wrap(Disease~.,ncol = 3)+
    theme_bw()  + 
    geom_vline(xintercept = 0,linetype="dotted", colour = "red")+
    theme(axis.ticks = element_blank(),  
          panel.grid.minor = element_blank(), 
          axis.line = element_line(colour = "black"), 
          axis.text = element_text(face="bold",size = 12, colour = "black"),
          axis.title = element_text(face="bold",size = 12),
          strip.text = element_text(face="bold",size = 12), 
          panel.spacing.x = unit(0.6,units = "cm"), 
          axis.title.y = element_blank(), 
          legend.position = "none", 
          plot.title = element_text(hjust = 0.5))
  if(!is.null(xlim)){
    p <- p + coord_cartesian(xlim = xlim)
  }
  print(p)
}

## change names for traits
change_annot_names <- function(annot_list){
  annot_list <- gsub("^CN$","iN-Glut OCR", annot_list)
  annot_list <- gsub("^DN$","iN-DN OCR", annot_list)
  annot_list <- gsub("^GA$","iN-GA OCR", annot_list)
  annot_list <- gsub("^ips$","iPSC OCR", annot_list)
  annot_list <- gsub("^NSC$","NPC OCR", annot_list)
  annot_list <- gsub("^ASoC_glut$","iN-Glut ASoC", annot_list)
  annot_list <- gsub("^ASoC_npc$","NPC ASoC", annot_list)
  return(annot_list)
}

## combine S-LDSC enrichment results across traits
combine_sldsc_traits <- function(trait_name_list, dir_results, baseline){
  registerDoParallel(cores = 10)
  
  result_sLDSC <- foreach(trait = trait_name_list, .combine = rbind)%dopar%{
    sldsc_results <- read.table(paste0(dir_results,"/", trait, "/", baseline, "/", trait,"_", "CN_all_peaks.narrowPeak.cleaned.hg19.merged", "_", baseline, ".results"), header = T, stringsAsFactors = F)
    sldsc.CN <- sldsc_results[sldsc_results$Category == "L2_1",]
    sldsc.CN$Category <- "CN"
    
    sldsc_results <- read.table(paste0(dir_results,"/", trait, "/", baseline, "/", trait,"_", "DN_all_peaks.narrowPeak.cleaned.hg19.merged", "_", baseline, ".results"), header = T, stringsAsFactors = F)
    sldsc.DN <- sldsc_results[sldsc_results$Category == "L2_1",]
    sldsc.DN$Category <- "DN"
    
    sldsc_results <- read.table(paste0(dir_results,"/", trait, "/", baseline, "/", trait,"_", "GA_all_peaks.narrowPeak.cleaned.hg19.merged", "_", baseline, ".results"), header = T, stringsAsFactors = F)
    sldsc.GA <- sldsc_results[sldsc_results$Category == "L2_1",]
    sldsc.GA$Category <- "GA"
    
    sldsc_results <- read.table(paste0(dir_results,"/", trait, "/", baseline, "/", trait,"_", "ips_all_peaks.narrowPeak.cleaned.hg19.merged", "_", baseline, ".results"), header = T, stringsAsFactors = F)
    sldsc.ips <- sldsc_results[sldsc_results$Category == "L2_1",]
    sldsc.ips$Category <- "ips"
    
    sldsc_results <- read.table(paste0(dir_results,"/", trait, "/", baseline, "/", trait,"_", "NSC_all_peaks.narrowPeak.cleaned.hg19.merged", "_", baseline, ".results"), header = T, stringsAsFactors = F)
    sldsc.NSC <- sldsc_results[sldsc_results$Category == "L2_1",]
    sldsc.NSC$Category <- "NSC"
    
    sldsc_results <- read.table(paste0(dir_results,"/", trait, "/", baseline, "/", trait,"_", "ASoC_glut_anno_hg19", "_", baseline, ".results"), header = T, stringsAsFactors = F)
    sldsc.ASoC_glut <- sldsc_results[sldsc_results$Category == "L2_1",]
    sldsc.ASoC_glut$Category <- "ASoC_glut"
    
    sldsc_results <- read.table(paste0(dir_results,"/", trait, "/", baseline, "/", trait,"_", "ASoC_npc_anno_hg19", "_", baseline, ".results"), header = T, stringsAsFactors = F)
    sldsc.ASoC_npc <- sldsc_results[sldsc_results$Category == "L2_1",]
    sldsc.ASoC_npc$Category <- "ASoC_npc"
    
    sldsc.combined <- rbind(sldsc.CN, sldsc.DN, sldsc.GA, sldsc.ips, sldsc.NSC, sldsc.ASoC_glut, sldsc.ASoC_npc)
    sldsc.combined <- cbind(Disease = trait, sldsc.combined)
    sldsc.combined
  }
  return(result_sLDSC)
}

```

```{r}
baseline <- "baselineLDv1.1"

dir_results <- "/project2/xinhe/kevinluo/ldsc/results/sLDSC_neuron_ATACseq_examples/"
result_sLDSC <- combine_sldsc_traits(trait_name_list, dir_results, baseline)

result_sLDSC$Category <- change_annot_names(result_sLDSC$Category)

result_sLDSC$Category <- factor(result_sLDSC$Category, levels = rev(annot_list) )
result_sLDSC$Color <- factor(result_sLDSC$Category, levels = annot_list)

```

#### Enrichment
```{r}
DT::datatable(format(result_sLDSC[,1:7], digits = 2), options = list(scrollX = TRUE, keys = TRUE, pageLength = length(annot_list)),rownames = F)
```

```{r}
ggplot_enrichment(result_sLDSC, title = "", xlim = c(0,50))
```


#### log2 Enrichment
```{r}
ggplot_log2_enrichment(result_sLDSC, title = "")
```

