---
title: "S-LSDC pipeline to estimate partitioned heritability and enrichment by functional annotations"
author: "Kaixuan Luo"
date: "2020-05-24"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

## Install LD Score Regression (LDSC) software
Please see the detail instructions: 
LD Score Regression (LDSC)
https://github.com/bulik/ldsc

### Download `ldsc` software
```{bash, eval=FALSE}
git clone https://github.com/bulik/ldsc.git
cd ldsc
```

### Create a conda environment with LDSC's dependencies
You might need to update numpy (and other packages) to a newer version
```{bash, eval=FALSE}
conda env create --file environment.yml
```

### Activate the conda environment with LDSC's dependencies
```{bash, eval=FALSE}
conda activate ldsc
```

### Test installation
If these commands fail with an error, then something as gone wrong during the installation process.
```{bash, eval=FALSE}
cd ldsc

python ./ldsc.py -h
python ./munge_sumstats.py -h
```

## `ldsc` Wiki
The [wiki](https://github.com/bulik/ldsc/wiki) has tutorials on estimating LD Score, heritability, genetic correlation and the LD Score regression intercept and partitioned heritability.

## Tutorials for partitioning heritability using S-LDSC (stratified LD score regression)
* [Partitioned Heritability](https://github.com/bulik/ldsc/wiki/Partitioned-Heritability)

* [Partitioned Heritability from Continuous Annotations](https://github.com/bulik/ldsc/wiki/Partitioned-Heritability-from-Continuous-Annotations)

## `ldsc` FAQ
Common issues are described in the [FAQ](https://github.com/bulik/ldsc/wiki/FAQ)

## References
* Partitioned heritability: 
Finucane, HK, et al. Partitioning heritability by functional annotation using genome-wide association summary statistics. Nature Genetics, 2015. 

* Stratified heritability using continuous annotation:
Gazal, S, et al. Linkage disequilibriumâ€“dependent architecture of human complex traits shows action of negative selection. Nature Genetics, 2017.

## Download LDSC reference data
You may need to download some of the following datasets:

Most of the data can be downloaded from the Price lab [LDSCORE](https://data.broadinstitute.org/alkesgroup/LDSCORE/) website

#### Download the baseline model LD scores
* [Readme](https://data.broadinstitute.org/alkesgroup/LDSCORE/readme_baseline_versions) of different versions of baseline models

* 1000G Phase3 baseline model v1.1
```{bash, eval=FALSE}
wget https://data.broadinstitute.org/alkesgroup/LDSCORE/1000G_Phase3_baseline_v1.1_ldscores.tgz
mkdir 1000G_Phase3_baseline_v1.1_ldscores
tar -xvzf 1000G_Phase3_baseline_v1.1_ldscores.tgz -C 1000G_Phase3_baseline_v1.1_ldscores
```

* 1000G Phase3 baselineLD model v1.1
```{bash, eval=FALSE}
wget https://data.broadinstitute.org/alkesgroup/LDSCORE/1000G_Phase3_baselineLD_v1.1_ldscores.tgz
mkdir 1000G_Phase3_baselineLD_v1.1_ldscores
tar -xvzf 1000G_Phase3_baselineLD_v1.1_ldscores.tgz -C 1000G_Phase3_baselineLD_v1.1_ldscores
```

* 1000G Phase3 baselineLD model v2.2
```{bash, eval=FALSE}
wget https://data.broadinstitute.org/alkesgroup/LDSCORE/1000G_Phase3_baselineLD_v2.2_ldscores.tgz
mkdir 1000G_Phase3_baselineLD_v2.2_ldscores
tar -xvzf 1000G_Phase3_baselineLD_v2.2_ldscores.tgz -C 1000G_Phase3_baselineLD_v2.2_ldscores
```

#### Download regression weights
```{bash, eval=FALSE}
# wget https://data.broadinstitute.org/alkesgroup/LDSCORE/weights_hm3_no_hla.tgz
# tar -xvzf weights_hm3_no_hla.tgz

# European of Phase 3 of 1000 Genomes
wget https://data.broadinstitute.org/alkesgroup/LDSCORE/1000G_Phase3_weights_hm3_no_MHC.tgz
tar -xvzf 1000G_Phase3_weights_hm3_no_MHC.tgz
```

#### Download allele frequencies (European of Phase 3 of 1000 Genomes)
```{bash, eval=FALSE}
wget https://data.broadinstitute.org/alkesgroup/LDSCORE/1000G_Phase3_frq.tgz
tar -xvzf 1000G_Phase3_frq.tgz
```

The authors recommend only keeping HapMap3 SNPs. You can download the HapMap3 related files:

#### Download 1000 genomes reference genotypes at HapMap3 loci
```{bash, eval=FALSE}
wget https://data.broadinstitute.org/alkesgroup/LDSCORE/1000G_Phase3_plinkfiles.tgz
tar -xvzf 1000G_Phase3_plinkfiles.tgz
```

#### Download HapMap3 SNPs
```{bash, eval=FALSE}
wget https://data.broadinstitute.org/alkesgroup/LDSCORE/hapmap3_snps.tgz
tar -xvzf hapmap3_snps.tgz
```

#### Download a concatenated list of HapMap3 SNPs
```{bash, eval=FALSE}
wget https://data.broadinstitute.org/alkesgroup/LDSCORE/w_hm3.snplist.bz2
bzip2 -d w_hm3.snplist.bz2

# Extract the list of HapMap 3 SNPs rsIDs
awk '{if ($1!="SNP") {print $1} }' w_hm3.snplist > listHM3.txt

```

## Compute LD score from functional annotations
* `ldsc` wiki ["LD-Score-Estimation-Tutorial"](https://github.com/bulik/ldsc/wiki/LD-Score-Estimation-Tutorial)

### Compute LD scores for binary annotations
* Step 1: prepare annotation in UCSC [BED](https://genome.ucsc.edu/FAQ/FAQformat.html#format1) format
* Step 2: compute LD scores using annotation BED files

* Example scripts:
`thin-annot` annotation format using `make_annot.py`
```{bash, eval=FALSE}
for chrom in {1..22}
do
  echo ${chrom}

  ## Step 1: Creating an annot file
  echo "Make ldsc-friendly annotation files for ${ANNOT}.bed"
  python make_annot.py \
  --bed-file ${ANNOT}.bed \
  --bimfile 1000G_EUR_Phase3_plink/1000G.EUR.QC.${chrom}.bim \
  --annot-file ${ANNOT}.${chrom}.annot.gz

  ## Step 2: Computing LD scores with an annot file
  echo "Computing LD scores with the annot file ${ANNOT}.${chrom}.annot.gz"
  python ldsc.py \
  --l2 \
  --bfile 1000G_EUR_Phase3_plink/1000G.EUR.QC.${chrom} \
  --print-snps listHM3.txt \
  --ld-wind-cm 1 \
  --annot ${ANNOT}.${chrom}.annot.gz \
  --thin-annot \
  --out ${ANNOT}.${chrom}
done
```

full annotation or `thin-annot` format using my own [code](../code/make_ldsc_binary_annot.R)
```{bash, eval=FALSE}
for chrom in {1..22}
do
  echo ${chrom}

  ## Step 1: Creating an annot file
  echo "Make ldsc-friendly annotation files for ${ANNOT}.bed"
  Rscript code/make_ldsc_binary_annot.R \
  ${ANNOT}.bed \
  1000G_EUR_Phase3_plink/1000G.EUR.QC.${chrom}.bim \
  ${ANNOT}.${chrom}.annot.gz "full-annot"

  ## Step 2: Computing LD scores with an annot file
  echo "Computing LD scores with the annot file ${ANNOT}.${chrom}.annot.gz"
  python ldsc.py \
  --l2 \
  --bfile 1000G_EUR_Phase3_plink/1000G.EUR.QC.${chrom} \
  --print-snps listHM3.txt \
  --ld-wind-cm 1 \
  --annot ${ANNOT}.${chrom}.annot.gz \
  --out ${ANNOT}.${chrom}
done
```

### Compute LD scores for continuous annotations
* Step 1: prepare annotation in UCSC [BED](https://genome.ucsc.edu/FAQ/FAQformat.html#format1) format
* Step 2: compute LD scores using annotation BED files

* Example script (full annotation format using my own [code](../code/make_ldsc_continuous_annot.R) )
```{bash, eval=FALSE}

for chrom in {1..22}
do
  echo ${chrom}

  ## Step 1: Creating an annot file (using make_ldsc_continuous_annot.R)
  echo "Make ldsc-friendly annotation files for ${ANNOT}.bed"
  Rscript code/make_ldsc_continuous_annot.R \
  ${ANNOT}.bed \
  1000G_EUR_Phase3_plink/1000G.EUR.QC.${chrom}.bim \
  ${ANNOT}.${chrom}.annot.gz "full-annot"

  ## Step 2: Computing LD scores with an annot file
  echo "Computing LD scores with the annot file ${ANNOT}.${chrom}.annot.gz"
  python ldsc.py \
  --l2 \
  --bfile 1000G_EUR_Phase3_plink/1000G.EUR.QC.${chrom} \
  --print-snps listHM3.txt \
  --ld-wind-cm 1 \
  --annot ${ANNOT}.${chrom}.annot.gz \
  --out ${ANNOT}.${chrom}

done

```

## Partitioned heritability (binary annotation)
* `ldsc` wiki ["Partitioned-Heritability"](https://github.com/bulik/ldsc/wiki/Partitioned-Heritability)

### Prepare GWAS summary stats in LDSC `.sumstats` format
* Convert GWAS summary stats to the LDSC `.sumstats` format using `munge_sumstats.py`
* `ldsc` wiki ["Summary-Statistics-File-Format"](https://github.com/bulik/ldsc/wiki/Partitioned-Heritability)

* Some of the processed GWAS summary stats can be found on RCC: 
`/project2/xinhe/kevinluo/GWAS/GWAS_summary_stats/GWAS_collection/ldsc_format/`

### Run ldsc on your GWAS summary statistics using baseline-LD model annotations and your new annotation
* Compute the annotation conditional on baselineLD model: controlling for the annotation categories of the full baselineLD model, using a comma-separated list of annotation file prefixes with `--ref-ld-chr`
```{bash, eval=FALSE}
python ldsc.py \
--h2 ${TRAIT}.sumstats.gz \
--ref-ld-chr baselineLD.,${ANNOT}. \
--frqfile-chr 1000G_Phase3_frq/1000G.EUR.QC. \
--w-ld-chr 1000G_Phase3_weights_hm3_no_MHC/weights.hm3_noMHC. \
--overlap-annot --print-cov --print-coefficients --print-delete-vals \
--out ${TRAIT}_${ANNOT}_baselineLD
```

* Joint model: you could include multiple annotations file prefixes to run multiple annotations jointly 
```{bash, eval=FALSE}
python ldsc.py \
--h2 ${TRAIT}.sumstats.gz \
--ref-ld-chr baselineLD.,${ANNOT_1}.,${ANNOT_2}. \
--frqfile-chr 1000G_Phase3_frq/1000G.EUR.QC. \
--w-ld-chr 1000G_Phase3_weights_hm3_no_MHC/weights.hm3_noMHC. \
--overlap-annot --print-cov --print-coefficients --print-delete-vals \
--out ${TRAIT}_joint_baselineLD
```

## Partitioned heritability (continuous annotation)
* `ldsc` allows taking continuous annotations as inputs for both --l2 and --h2 options. The pipeline is similar to that using binary annotation. However, some result outputs of --h2 option are not meaningful anymore. Computing the proportion of heritability explained by each quantile of a continuous annotation provides a more intuitive interpretation of the magnitude of a continuous annotation effects. You can use their R script `quantile_h2g.r` and follow their wiki tutorial to compute the proportion of heritability explained by each quintile.

* Please follow the `ldsc` wiki ["Partitioned Heritability from Continuous Annotations"](https://github.com/bulik/ldsc/wiki/Partitioned-Heritability-from-Continuous-Annotations)

